// @kuping/oto_component/lib/business/BusinessNotice/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var _common = require("../../utils/common");
require("./index.css");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var BusinessNotice = /*#__PURE__*/function (_Component) {
  _inheritsLoose(BusinessNotice, _Component);
  function BusinessNotice() {
    var _this;
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _Component.call.apply(_Component, [this].concat(args)) || this;
    _this.scrollTime = null;
    _this.LEN = 200;
    _this.x = 0;
    _this.textEl = void 0;
    _this.wrapEl = void 0;
    return _this;
  }
  var _proto = BusinessNotice.prototype;
  _proto.componentDidMount = function componentDidMount() {
    var _this2 = this;
    var data = this.props.data;
    var _ref = data || {},
      dataSource = _ref.dataSource;
    var _ref2 = dataSource || {},
      name = _ref2.name;
    if (typeof name === 'string' && name.length > 20) {
      setTimeout(function () {
        return _this2.initCloneDom();
      }, 1500);
    }
  };
  _proto.componentWillUnmount = function componentWillUnmount() {
    if (this.scrollTime) clearInterval(this.scrollTime);
  };
  _proto.roll = function roll() {
    var _this3 = this;
    var tag1 = this.textEl;
    if (!tag1) return;
    var tag2 = tag1.nextSibling;
    if (!tag2) return;
    var fun = function fun() {
      tag1.style.left = _this3.x + "px";
      tag2.style.left = _this3.x + _this3.LEN + "px";
      _this3.x = _this3.x - 1;
      if (_this3.x + _this3.LEN === 0) {
        _this3.x = 0;
        tag1.style.transition = 'none';
        tag2.style.transition = 'none';
        setTimeout(function () {
          tag1.style.transition = 'transform';
          tag2.style.transition = 'transform';
        }, 30);
      }
    };
    if (this.scrollTime) {
      clearInterval(this.scrollTime);
    }
    this.scrollTime = setInterval(fun, 30);
  };
  _proto.initCloneDom = function initCloneDom() {
    var el = this.textEl;
    var el2 = el.cloneNode(true);
    this.LEN = el.clientWidth + this.wrapEl.clientWidth;
    el2.style.left = this.x + this.LEN + "px";
    el.parentElement.appendChild(el2);
    this.roll();
  };
  _proto.handleClick = function handleClick(config) {
    var _this$props = this.props,
      onClick = _this$props.onClick,
      data = _this$props.data;
    var _ref3 = data || {},
      _ref3$basicAttribute = _ref3.basicAttribute,
      basicAttribute = _ref3$basicAttribute === void 0 ? {} : _ref3$basicAttribute;
    var row = _extends({}, config, basicAttribute.monitor || {});
    row.componentName = '公告';
    (0, _common.globalClick)(row, onClick);
  };
  _proto.render = function render() {
    var _this4 = this;
    var data = this.props.data;
    var _ref4 = data || {},
      basicAttribute = _ref4.basicAttribute,
      dataSource = _ref4.dataSource;
    var _ref5 = basicAttribute || {},
      relationStyle = _ref5.relationStyle;
    var _ref6 = dataSource || {},
      imgUrl = _ref6.imgUrl,
      name = _ref6.name;
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: "template-notify-container"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "notify-container"
    }, relationStyle === 1 ? imgUrl ? /*#__PURE__*/_react["default"].createElement("img", {
      src: imgUrl,
      alt: "",
      className: "notify-img"
    }) : /*#__PURE__*/_react["default"].createElement("div", {
      className: "notify-lab"
    }, "\u516C\u544A") : /*#__PURE__*/_react["default"].createElement("div", {
      style: {
        height: '0.4rem',
        width: '0.01rem'
      }
    }), /*#__PURE__*/_react["default"].createElement("div", {
      className: "notify-text-wrapper",
      ref: function ref(el) {
        return _this4.wrapEl = el;
      },
      onClick: function onClick() {
        return _this4.handleClick(dataSource || {});
      }
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "notify-text",
      ref: function ref(el) {
        return _this4.textEl = el;
      }
    }, name))));
  };
  return BusinessNotice;
}(_react.Component);
BusinessNotice.defaultProps = {
  attr: {
    name: '公告'
  }
};
var _default = BusinessNotice;
exports["default"] = _default;
