// @kp-react-visual-design/design/lib/components/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Drag", {
  enumerable: true,
  get: function get() {
    return _Drag.default;
  }
});
Object.defineProperty(exports, "FileManage", {
  enumerable: true,
  get: function get() {
    return _FileManage.default;
  }
});
Object.defineProperty(exports, "HotArea", {
  enumerable: true,
  get: function get() {
    return _HotArea.default;
  }
});
Object.defineProperty(exports, "LinkCoupon", {
  enumerable: true,
  get: function get() {
    return _LinkCoupon.default;
  }
});
Object.defineProperty(exports, "LinkGoods", {
  enumerable: true,
  get: function get() {
    return _LinkGoods.default;
  }
});
Object.defineProperty(exports, "LinkRecommendId", {
  enumerable: true,
  get: function get() {
    return _LinkRecommendId.default;
  }
});
Object.defineProperty(exports, "LinksButton", {
  enumerable: true,
  get: function get() {
    return _LinksButton.default;
  }
});
Object.defineProperty(exports, "TableSearch", {
  enumerable: true,
  get: function get() {
    return _TableSearch.default;
  }
});

var _Drag = _interopRequireDefault(require("./Drag"));

var _HotArea = _interopRequireDefault(require("./HotArea"));

var _FileManage = _interopRequireDefault(require("./FileManage"));

var _LinksButton = _interopRequireDefault(require("./LinksButton"));

var _LinkCoupon = _interopRequireDefault(require("./LinkCoupon"));

var _LinkGoods = _interopRequireDefault(require("./LinkGoods"));

var _LinkRecommendId = _interopRequireDefault(require("./LinkRecommendId"));

var _TableSearch = _interopRequireDefault(require("./TableSearch"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
