// @kuping/oto_component/lib/business/Items/Commodity.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var _common = require("../../utils/common");
var _Price = _interopRequireDefault(require("./Price/Price"));
var _Price2 = _interopRequireDefault(require("./Price/Price1"));
var _Price3 = _interopRequireDefault(require("./Price/Price2"));
var _Price4 = _interopRequireDefault(require("./Price/Price3"));
var _Price5 = _interopRequireDefault(require("./Price/Price4"));
var _IntegralPrice = _interopRequireDefault(require("./Price/IntegralPrice"));
require("./style/list.css");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var Commodity = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Commodity, _Component);
  function Commodity() {
    return _Component.apply(this, arguments) || this;
  }
  var _proto = Commodity.prototype;
  _proto.toCommodityDetail = function toCommodityDetail(path, code) {
    var _this$props$monitor = this.props.monitor,
      monitor = _this$props$monitor === void 0 ? {} : _this$props$monitor;
    var config = _extends({}, monitor, {
      name: code
    });
    (0, _common.baiduHmt)(config);
    if (!path || !path.trim('')) return;
    location.href = path + "?specificationCode=" + code + "&" + (0, _common.putParams)();
  };
  _proto.render = function render() {
    var _this = this;
    var _this$props = this.props,
      data = _this$props.data,
      _this$props$priceType = _this$props.priceType,
      priceType = _this$props$priceType === void 0 ? 2 : _this$props$priceType,
      customizePrice = _this$props.customizePrice,
      newLine = _this$props.newLine,
      tag = _this$props.tag;
    var saleTipText = this.props.saleTipText || '预估价';
    var _ref = data || {},
      commodityUrl = _ref.commodityUrl,
      commodityPic = _ref.commodityPic,
      commodityName = _ref.commodityName,
      originalPrice = _ref.originalPrice,
      activityGoodsVo = _ref.activityGoodsVo,
      commoditySpecifications = _ref.commoditySpecifications,
      integralPrice = _ref.integralPrice,
      displayCrossedPrice = _ref.displayCrossedPrice;
    var sellingPrice = activityGoodsVo ? activityGoodsVo.specialPrice : data.sellingPrice;
    var specificationCode = commoditySpecifications ? commoditySpecifications.specificationCode : '';
    var commoditySpecCode = activityGoodsVo ? activityGoodsVo.commoditySpecCode : specificationCode;
    var newIntegralPrice = activityGoodsVo ? activityGoodsVo.integralPrice : integralPrice;
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: "card",
      onClick: function onClick() {
        return _this.toCommodityDetail(commodityUrl, commoditySpecCode);
      },
      style: {
        display: 'block'
      }
    }, /*#__PURE__*/_react["default"].createElement("img", {
      src: commodityPic,
      alt: "",
      id: commodityPic,
      onError: function onError() {
        var imgEl = document.getElementById("" + commodityPic);
        if (!imgEl) return;
        imgEl.src = 'https://newboss.oss-cn-beijing.aliyuncs.com/pro/20220112/20220112185419WX20220112-185251%402x.png';
      }
    }), /*#__PURE__*/_react["default"].createElement("div", {
      className: "card-info"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "info-title"
    }, commodityName), customizePrice && tag && /*#__PURE__*/_react["default"].createElement("div", {
      className: "info-tag-name"
    }, tag), customizePrice ? /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, newIntegralPrice > 0 ? /*#__PURE__*/_react["default"].createElement(_IntegralPrice["default"], {
      price: sellingPrice,
      originPrice: originalPrice,
      saleTipText: saleTipText,
      integralPrice: newIntegralPrice,
      displayCrossedPrice: displayCrossedPrice
    }) : /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, (priceType === 1 || !priceType) && /*#__PURE__*/_react["default"].createElement(_Price["default"], {
      price: sellingPrice
    }), priceType === 2 && /*#__PURE__*/_react["default"].createElement(_Price5["default"], {
      price: sellingPrice,
      originPrice: originalPrice,
      saleTipText: saleTipText
    }))) : newIntegralPrice > 0 ? /*#__PURE__*/_react["default"].createElement(_IntegralPrice["default"], {
      price: sellingPrice,
      originPrice: originalPrice,
      saleTipText: saleTipText,
      integralPrice: newIntegralPrice,
      displayCrossedPrice: displayCrossedPrice
    }) : /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, priceType === 1 && /*#__PURE__*/_react["default"].createElement(_Price2["default"], {
      price: sellingPrice
    }), priceType === 2 && /*#__PURE__*/_react["default"].createElement(_Price3["default"], {
      price: sellingPrice,
      originPrice: originalPrice,
      saleTipText: saleTipText
    }), priceType === 3 && /*#__PURE__*/_react["default"].createElement(_Price4["default"], {
      price: sellingPrice,
      originPrice: originalPrice,
      saleTipText: saleTipText,
      newLine: newLine
    }))));
  };
  return Commodity;
}(_react.Component);
var _default = Commodity;
exports["default"] = _default;
