// @kuping/oto_component/lib/business/Items/Coupon.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var _common = require("../../utils/common");
var _qiang = _interopRequireDefault(require("./img/qiang.png"));
require("./style/list.css");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var Coupon = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Coupon, _Component);
  function Coupon(props) {
    var _this;
    _this = _Component.call(this, props) || this;
    _this.state = {};
    return _this;
  }
  var _proto = Coupon.prototype;
  _proto.goCardDetail = function goCardDetail(id, denominationId) {
    var _this$props$monitor = this.props.monitor,
      monitor = _this$props$monitor === void 0 ? {} : _this$props$monitor;
    var config = _extends({}, monitor, {
      name: id
    });
    (0, _common.baiduHmt)(config);
    var domainName = /192.168.|localhost|0.0.0.0/.test(location.hostname) ? 'https://dev-me.otosaas.com' : location.origin;
    location.href = domainName + "/kaquan/detail/" + id + "?denominationID=" + denominationId;
  };
  _proto.render = function render() {
    var _this2 = this;
    var data = this.props.data;
    var _ref = data || {},
      cardId = _ref.cardId,
      denominationId = _ref.denominationId,
      denominationImage = _ref.denominationImage,
      commodityPic = _ref.commodityPic,
      denominationTitle = _ref.denominationTitle,
      disAmount = _ref.disAmount,
      maxDiscount = _ref.maxDiscount,
      originalPrice = _ref.originalPrice,
      sale = _ref.sale;
    return /*#__PURE__*/_react["default"].createElement("div", {
      onClick: function onClick() {
        return _this2.goCardDetail(cardId, denominationId);
      }
    }, /*#__PURE__*/_react["default"].createElement("img", {
      src: denominationImage,
      alt: "",
      id: denominationImage,
      onError: function onError() {
        var imgEl = document.getElementById("" + commodityPic);
        if (!imgEl) return;
        imgEl.src = 'https://newboss.oss-cn-beijing.aliyuncs.com/pro/20220112/20220112185419WX20220112-185251%402x.png';
      }
    }), /*#__PURE__*/_react["default"].createElement("div", {
      className: "card-info"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "info-title"
    }, denominationTitle), /*#__PURE__*/_react["default"].createElement("div", {
      className: "tag",
      style: {
        borderWidth: '1px'
      }
    }, disAmount > 0 ? "\u7701" + disAmount + "\u5143" : '官方严选'), maxDiscount && maxDiscount < 10 ? /*#__PURE__*/_react["default"].createElement("div", {
      className: "tag",
      style: {
        borderWidth: '1px'
      }
    }, maxDiscount, "\u6298\u8D77") : /*#__PURE__*/_react["default"].createElement("span", null), /*#__PURE__*/_react["default"].createElement("div", {
      className: "info-card-price"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "price-current"
    }, /*#__PURE__*/_react["default"].createElement("span", null, "\xA5"), /*#__PURE__*/_react["default"].createElement("b", null, sale)), parseFloat(originalPrice) > parseFloat(sale) && /*#__PURE__*/_react["default"].createElement("s", {
      className: "price-original"
    }, "\xA5", originalPrice), /*#__PURE__*/_react["default"].createElement("img", {
      src: _qiang["default"],
      alt: ""
    }))));
  };
  return Coupon;
}(_react.Component);
var _default = Coupon;
exports["default"] = _default;
