// @kuping/oto_component/lib/business/Items/Shop.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var _const = require("../../utils/const");
var _common = require("../../utils/common");
var _poster_tag = _interopRequireDefault(require("./img/poster_tag.png"));
var _Rate = _interopRequireDefault(require("./Rate"));
require("./style/shop.css");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var Shop = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Shop, _Component);
  function Shop() {
    return _Component.apply(this, arguments) || this;
  }
  var _proto = Shop.prototype;
  _proto.toDetail = function toDetail(shopCode) {
    var _this$props = this.props,
      _this$props$monitor = _this$props.monitor,
      monitor = _this$props$monitor === void 0 ? {} : _this$props$monitor,
      appCode = _this$props.appCode;
    var config = _extends({}, monitor, {
      name: shopCode
    });
    (0, _common.baiduHmt)(config);
    var query = location.search.replace(/\?/g, '&');
    var domainName = '';
    if (/192.168.|localhost|0.0.0.0|dev-|test-/.test(location.hostname)) {
      domainName = "https://test-" + appCode + "-shopapp.otosaas.com";
    } else {
      domainName = "https://" + appCode + "-shopapp.otosaas.com";
    }
    location.href = domainName + "/shops/details?shopCode=" + shopCode + query;
  };
  _proto.render = function render() {
    var _this = this;
    var data = this.props.data;
    var _ref = data || {},
      listLogo = _ref.listLogo,
      intelligentWeight = _ref.intelligentWeight,
      shopName = _ref.shopName,
      status = _ref.status,
      shopScore = _ref.shopScore,
      areaName = _ref.areaName,
      categoryNameC1 = _ref.categoryNameC1,
      distance = _ref.distance,
      distanceUnit = _ref.distanceUnit,
      payDiscount = _ref.payDiscount,
      voucher = _ref.voucher,
      mealVoucher = _ref.mealVoucher,
      openStatus = _ref.openStatus,
      shopCode = _ref.shopCode;
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: "business-area-item",
      onClick: function onClick() {
        return _this.toDetail(shopCode);
      }
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "poster"
    }, /*#__PURE__*/_react["default"].createElement("img", {
      src: listLogo || 'https://newboss.oss-cn-beijing.aliyuncs.com/pro/20220112/20220112185419WX20220112-185251%402x.png',
      className: "poster-img",
      alt: "",
      id: shopCode,
      onError: function onError() {
        var imgEl = document.getElementById("" + shopCode);
        if (!imgEl) return;
        imgEl.src = 'https://newboss.oss-cn-beijing.aliyuncs.com/pro/20220112/20220112185419WX20220112-185251%402x.png';
      }
    }), intelligentWeight > 0 && /*#__PURE__*/_react["default"].createElement("img", {
      src: _poster_tag["default"],
      alt: "tag",
      className: "poster-tag"
    })), /*#__PURE__*/_react["default"].createElement("div", {
      className: "description"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "description-title"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "description-title-name"
    }, shopName), status === 4 && /*#__PURE__*/_react["default"].createElement("div", {
      className: "description-title-status"
    }, "\u6682\u505C\u8425\u4E1A"), openStatus === 2 && /*#__PURE__*/_react["default"].createElement("div", {
      className: "description-title-status rest"
    }, "\u4F11\u606F\u4E2D")), shopScore ? /*#__PURE__*/_react["default"].createElement("div", {
      className: "description-score"
    }, /*#__PURE__*/_react["default"].createElement(_Rate["default"], {
      value: shopScore
    }), /*#__PURE__*/_react["default"].createElement("div", {
      className: "count"
    }, shopScore.toFixed(1))) : /*#__PURE__*/_react["default"].createElement("div", {
      className: "description-area-info"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "store-tag"
    }, "\u6682\u65E0\u8BC4\u5206")), /*#__PURE__*/_react["default"].createElement("div", {
      className: "description-area-info"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "store-tag"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "area"
    }, areaName), /*#__PURE__*/_react["default"].createElement("div", {
      className: "store-type"
    }, categoryNameC1)), /*#__PURE__*/_react["default"].createElement("div", {
      className: "location"
    }, distance, distanceUnit)), payDiscount && !!JSON.parse(payDiscount).length && /*#__PURE__*/_react["default"].createElement("div", {
      className: "description-full-minus"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "buy"
    }, "\u4E70"), /*#__PURE__*/_react["default"].createElement("div", {
      className: "info"
    }, JSON.parse(payDiscount).map(function (item, index) {
      return "" + item.message + (JSON.parse(payDiscount).length !== index + 1 ? ',' : '');
    }))), voucher && !!JSON.parse(voucher).length && /*#__PURE__*/_react["default"].createElement("div", {
      className: "description-activity"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "coupon"
    }, "\u5238"), /*#__PURE__*/_react["default"].createElement("div", {
      className: "info"
    }, JSON.parse(voucher).map(function (item, index) {
      return "" + item.message + (JSON.parse(voucher).length !== index + 1 ? ',' : '');
    }))), mealVoucher && !!JSON.parse(mealVoucher).length && /*#__PURE__*/_react["default"].createElement("div", {
      className: "description-other"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "group"
    }, "\u56E2"), /*#__PURE__*/_react["default"].createElement("div", {
      className: "info"
    }, JSON.parse(mealVoucher).map(function (item, index) {
      return "" + item.message + (JSON.parse(mealVoucher).length !== index + 1 ? ',' : '');
    })))));
  };
  return Shop;
}(_react.Component);
Shop.defaultProps = {
  appCode: _const.defaultAppCode
};
var _default = Shop;
exports["default"] = _default;
