// @kuping/oto_component/lib/business/Items/Tickets.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var _common = require("../../utils/common");
require("./style/list.css");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var levelText = {
  无: {
    levelStr: 3,
    score: '7.5'
  },
  A: {
    levelStr: 1,
    score: '6.9'
  },
  AA: {
    levelStr: 2,
    score: '7.2'
  },
  AAA: {
    levelStr: 3,
    score: '7.8'
  },
  AAAA: {
    levelStr: 4,
    score: '8.2'
  },
  AAAAA: {
    levelStr: 5,
    score: '9.2'
  }
};
var Tickets = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Tickets, _Component);
  function Tickets() {
    return _Component.apply(this, arguments) || this;
  }
  var _proto = Tickets.prototype;
  _proto.goMenpiaoDetail = function goMenpiaoDetail(id) {
    var _this$props$monitor = this.props.monitor,
      monitor = _this$props$monitor === void 0 ? {} : _this$props$monitor;
    var config = _extends({}, monitor, {
      name: id
    });
    (0, _common.baiduHmt)(config);
    var domainName = /192.168.|localhost|0.0.0.0/.test(location.hostname) ? 'https://dev-me.otosaas.com' : location.origin;
    location.href = domainName + "/menpiao/details/" + id;
  };
  _proto.render = function render() {
    var _this = this;
    var data = this.props.data;
    var _ref = data || {},
      id = _ref.id,
      level = _ref.level,
      image = _ref.image,
      name = _ref.name,
      disName = _ref.disName,
      price = _ref.price,
      marketPrice = _ref.marketPrice;
    return /*#__PURE__*/_react["default"].createElement("div", {
      onClick: function onClick() {
        return _this.goMenpiaoDetail(id);
      }
    }, /*#__PURE__*/_react["default"].createElement("img", {
      src: image,
      alt: "",
      id: id,
      onError: function onError() {
        var imgEl = document.getElementById("" + id);
        if (!imgEl) return;
        imgEl.src = 'https://newboss.oss-cn-beijing.aliyuncs.com/pro/20220112/20220112185419WX20220112-185251%402x.png';
      }
    }), /*#__PURE__*/_react["default"].createElement("div", {
      className: "card-info"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "info-title"
    }, name), /*#__PURE__*/_react["default"].createElement("div", {
      className: "tag-info tag-menpiao"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "tag-wrap"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "score"
    }, levelText[level || '无'].score), /*#__PURE__*/_react["default"].createElement("div", {
      className: "tag",
      style: {
        borderWidth: '1px'
      }
    }, levelText[level || '无'].levelStr, "A\u666F\u533A")), /*#__PURE__*/_react["default"].createElement("div", {
      className: "distance"
    }, disName)), /*#__PURE__*/_react["default"].createElement("div", {
      className: "info-price"
    }, /*#__PURE__*/_react["default"].createElement("span", {
      className: "price-current"
    }, /*#__PURE__*/_react["default"].createElement("span", null, "\xA5"), /*#__PURE__*/_react["default"].createElement("b", null, price || 0)), marketPrice > price && /*#__PURE__*/_react["default"].createElement("s", {
      className: "price-original"
    }, "\xA5", marketPrice))));
  };
  return Tickets;
}(_react.Component);
var _default = Tickets;
exports["default"] = _default;
