// @kuping/oto_component/lib/business/Carousel/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var _antdMobileV = require("antd-mobile-v5");
var _common = require("../../utils/common");
var _CustomImage = _interopRequireDefault(require("../CustomImage"));
var _default2 = require("./default");
require("./index.css");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var defaultImg = 'https://bosscdn.otosaas.com/test/20221108/20221108160142702%2A280.png?path=0,650,663';
var Carousel = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Carousel, _Component);
  function Carousel() {
    return _Component.apply(this, arguments) || this;
  }
  var _proto = Carousel.prototype;
  _proto.imgLoad = function imgLoad(current) {
    // 图片高度是自适应的 等待图片加载出来之后 让滚动条滚动至顶部
    // const containerEl = document.getElementsByClassName('pindao')[0];
    // if (!current) return;
    var bannerEl = current.currentTarget.closest('.template-banner');
    // if (containerEl) {
    //   containerEl.scrollTop = 0;
    // }
    try {
      var naturalHeight = current.currentTarget.naturalHeight;
      current.currentTarget.style.width = '100%';
      current.currentTarget.style.height = naturalHeight / 100 + "rem";
      if (bannerEl) {
        bannerEl.style.height = naturalHeight / 100 + "rem";
      }
    } catch (error) {
      console.log('没有获取到图片原始高度：', error);
    }
  };
  _proto.handleClick = function handleClick(config) {
    var _this$props = this.props,
      onClick = _this$props.onClick,
      componentName = _this$props.componentName;
    config.componentName = componentName || '轮播';
    (0, _common.globalClick)(config, onClick);
  };
  _proto.render = function render() {
    var _this = this;
    var _this$props2 = this.props,
      swiperType = _this$props2.swiperType,
      data = _this$props2.data;
    var _ref = data || {},
      _dataSource = _ref.dataSource,
      basicAttribute = _ref.basicAttribute;
    var dataSource = _dataSource && _dataSource.length ? _dataSource : _default2.defaultDataSource;
    var _ref2 = basicAttribute || {},
      monitor = _ref2.monitor;
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: swiperType === 1 ? 'template-banner' : 'template-banner banner-direct',
      style: {
        padding: swiperType === 1 ? '0 0.243rem' : '0'
      }
    }, dataSource.length === 1 ? /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(_CustomImage["default"], {
      onClick: function onClick() {
        return _this.handleClick(_extends({}, dataSource[0], monitor || {}));
      },
      alt: "",
      src: dataSource[0].imgUrl || defaultImg,
      style: {
        height: '2.8rem',
        width: '100%'
      },
      onLoad: function onLoad(e) {
        return _this.imgLoad(e);
      }
    })) : /*#__PURE__*/_react["default"].createElement(_antdMobileV.Swiper, {
      rubberband: false,
      loop: true,
      autoplay: true,
      indicatorProps: {
        style: {
          '--dot-color': 'rgba(255, 255, 255, 0.6)',
          '--active-dot-color': 'rgba(255, 255, 255, 1)',
          '--dot-size': '.12rem',
          '--active-dot-size': '.24rem',
          '--dot-border-radius': '50%',
          '--active-dot-border-radius': '.8rem'
        }
      }
    }, dataSource.map(function (banner, index) {
      return /*#__PURE__*/_react["default"].createElement(_antdMobileV.Swiper.Item, {
        key: index
      }, /*#__PURE__*/_react["default"].createElement("div", {
        onClick: function onClick() {
          return _this.handleClick(_extends({}, banner, monitor || {}));
        },
        style: {
          display: 'flex',
          justifyContent: 'center',
          overflow: 'hidden'
        }
      }, /*#__PURE__*/_react["default"].createElement(_CustomImage["default"], {
        alt: "",
        src: banner.imgUrl || defaultImg,
        onLoad: function onLoad(e) {
          return _this.imgLoad(e);
        },
        style: {
          height: '2.8rem',
          width: '100%',
          display: 'block'
        }
      })));
    })));
  };
  return Carousel;
}(_react.Component);
Carousel.defaultProps = {
  data: {
    dataSource: [],
    basicAttribute: {}
  },
  swiperType: 1,
  attr: {
    name: '轮播'
  }
};
var _default = Carousel;
exports["default"] = _default;
