// @kp-react-visual-design/design/es/components/TableSearch/index.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import { Row, Input, Select, Button, DatePicker } from 'antd';
import { StyledCol, ButtonContainer, Container } from './index.styled';
var components = {
  Input: Input,
  Select: Select,
  DatePicker: DatePicker,
  RangePicker: DatePicker.RangePicker
};

var SearchInputs = function SearchInputs(_ref) {
  var options = _ref.options,
      loading = _ref.loading,
      onChange = _ref.onChange,
      onSearch = _ref.onSearch,
      onReset = _ref.onReset,
      values = _ref.values;

  var handleChange = function handleChange(name, componentName, e) {
    if (componentName === 'Input') {
      onChange === null || onChange === void 0 ? void 0 : onChange(_defineProperty({}, name, e.target.value));
    } else {
      onChange === null || onChange === void 0 ? void 0 : onChange(_defineProperty({}, name, e));
    }
  };

  return /*#__PURE__*/React.createElement(Container, null, /*#__PURE__*/React.createElement(Row, null, options.map(function (item, index) {
    var Component = components[item.componentName];
    return /*#__PURE__*/React.createElement(StyledCol, {
      span: 8,
      key: index
    }, /*#__PURE__*/React.createElement("span", {
      className: "label"
    }, item.label, ":"), /*#__PURE__*/React.createElement(Component, {
      defaultValue: item.defaultValue,
      placeholder: item.placeholder,
      onChange: function onChange(e) {
        return handleChange(item.name, item.componentName, e);
      },
      style: {
        flex: '1 0 0'
      },
      options: item.options,
      value: item.componentName === 'RangePicker' ? values.date : values[item.name]
    }));
  })), !!options.length && /*#__PURE__*/React.createElement(ButtonContainer, null, /*#__PURE__*/React.createElement(Button, {
    onClick: onReset,
    style: {
      marginRight: 10
    }
  }, "\u91CD\u7F6E"), /*#__PURE__*/React.createElement(Button, {
    type: "primary",
    loading: loading,
    disabled: loading,
    onClick: onSearch
  }, "\u67E5\u8BE2")));
};

SearchInputs.defaultProps = {
  options: [],
  loading: false,
  onChange: function onChange() {}
};
export default SearchInputs;
