// @boluome/filemanage-react-component-library/es/FileManage/BasisLayout/index.js
var _excluded = ["defalutActive", "sidebar"],
    _excluded2 = ["settings", "currentSelectedList", "onResetFetch", "searchValue", "onSearch", "dataSource", "beforeUpload", "activeIndex", "onChangeState"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { useState, useContext, useMemo } from 'react';
import { ModalBasisLayout, ModalContainer, ModalSidebar, ModalContent, ModalActionArea, FilePreviewStryled } from "./styled";
import { KpIconGroup, KpIcon, KpInput, KpUpload } from "../../componnets";
import ModalFilesContent from "./ModalFilesContent";
import { FileManageContext } from "./..";

var BasisLayout = function BasisLayout() {
  var _useContext = useContext(FileManageContext),
      _useContext$settings = _useContext.settings,
      defalutActive = _useContext$settings.defalutActive,
      width = _useContext$settings.sidebar.width,
      restSettings = _objectWithoutProperties(_useContext$settings, _excluded),
      currentSelectedList = _useContext.currentSelectedList,
      onResetFetch = _useContext.onResetFetch,
      searchValue = _useContext.searchValue,
      onSearch = _useContext.onSearch,
      dataSource = _useContext.dataSource,
      beforeUpload = _useContext.beforeUpload,
      activeIndex = _useContext.activeIndex,
      onChangeState = _useContext.onChangeState,
      rest = _objectWithoutProperties(_useContext, _excluded2);

  var _useState = useState(defalutActive),
      _useState2 = _slicedToArray(_useState, 2),
      active = _useState2[0],
      setActive = _useState2[1];

  var isShowBackIcon = useMemo(function () {
    var _breadcrumb = JSON.parse(window.localStorage.getItem('breadcrumb') || '');

    return _breadcrumb[_breadcrumb.length - 1] !== 0;
  }, [window.localStorage.getItem('breadcrumb')]);

  var handleIconChange = function handleIconChange(index) {
    setActive(index);
  };

  var handleBack = function handleBack() {
    var breadcrumb = JSON.parse(window.localStorage.getItem('breadcrumb') || '');
    breadcrumb.splice(breadcrumb.length - 1, 1);
    window.localStorage.setItem('breadcrumb', JSON.stringify(breadcrumb));
    onChangeState({
      menuVisible: false
    });
    onResetFetch();
  };

  var handleClearSelect = function handleClearSelect() {
    onChangeState({
      currentSelectedList: []
    });
  };

  var handleUploadChange = function handleUploadChange(filelist) {
    console.log(filelist);
  };

  return /*#__PURE__*/React.createElement(ModalBasisLayout, null, /*#__PURE__*/React.createElement(ModalContainer, null, /*#__PURE__*/React.createElement(ModalSidebar, {
    style: {
      width: width
    }
  }, /*#__PURE__*/React.createElement(KpUpload, {
    type: "primary",
    style: {
      width: '100%'
    },
    onChange: handleUploadChange,
    beforeUpload: beforeUpload
  }, /*#__PURE__*/React.createElement("i", {
    className: "iconfont icon-jiahao"
  }), /*#__PURE__*/React.createElement("span", null, "\u4E0A\u4F20")), activeIndex !== null && /*#__PURE__*/React.createElement(FilePreviewStryled, null, /*#__PURE__*/React.createElement("div", {
    className: "file-preview"
  }, dataSource[activeIndex].type === 1 ? /*#__PURE__*/React.createElement("i", {
    className: "iconfont icon-wenjianjia"
  }) : /*#__PURE__*/React.createElement("img", {
    src: dataSource[activeIndex].url,
    alt: ""
  })), /*#__PURE__*/React.createElement("div", {
    className: "file-id"
  }, "ID\uFF1A", dataSource[activeIndex].id), /*#__PURE__*/React.createElement("div", {
    className: "file-name"
  }, "\u540D\u79F0\uFF1A", dataSource[activeIndex].name), dataSource[activeIndex].suffix && /*#__PURE__*/React.createElement("div", {
    className: "file-suffix"
  }, "\u683C\u5F0F\uFF1A", dataSource[activeIndex].suffix), dataSource[activeIndex].size && /*#__PURE__*/React.createElement("div", {
    className: "file-size"
  }, "\u5927\u5C0F\uFF1A", dataSource[activeIndex].size), dataSource[activeIndex].createdAt && /*#__PURE__*/React.createElement("div", {
    className: "create-at"
  }, "\u65E5\u671F\uFF1A", dataSource[activeIndex].createdAt), dataSource[activeIndex].createdBy && /*#__PURE__*/React.createElement("div", {
    className: "create-author"
  }, "\u64CD\u4F5C\u4EBA\uFF1A", dataSource[activeIndex].createdBy), dataSource[activeIndex].updatedAt && /*#__PURE__*/React.createElement("div", {
    className: "update-at"
  }, "\u66F4\u65B0\u65E5\u671F\uFF1A", dataSource[activeIndex].updatedAt), dataSource[activeIndex].updatedBy && /*#__PURE__*/React.createElement("div", {
    className: "update-author"
  }, "\u64CD\u4F5C\u4EBA\uFF1A", dataSource[activeIndex].updatedBy))), /*#__PURE__*/React.createElement(ModalContent, null, /*#__PURE__*/React.createElement(ModalActionArea, null, isShowBackIcon && /*#__PURE__*/React.createElement(KpIcon, {
    type: "icon-zuo",
    onClick: handleBack,
    shape: "circle",
    style: {
      marginRight: 8
    }
  }), /*#__PURE__*/React.createElement(KpIconGroup, {
    defalutIndex: active,
    onChange: handleIconChange
  }, /*#__PURE__*/React.createElement(KpIcon, {
    shape: "circle",
    type: "icon-liebiao1"
  }), /*#__PURE__*/React.createElement(KpIcon, {
    shape: "circle",
    type: "icon-liebiao2"
  })), !!currentSelectedList.length && /*#__PURE__*/React.createElement(KpIcon, {
    shape: "circle",
    type: "icon-cuohaoyuan",
    style: {
      marginRight: 8
    },
    onClick: handleClearSelect
  }), /*#__PURE__*/React.createElement(KpInput, {
    width: "100%",
    style: {
      marginRight: 8
    },
    value: searchValue,
    onChange: function onChange(value) {
      return onChangeState({
        searchValue: value
      });
    },
    onPressEnter: onSearch
  }, /*#__PURE__*/React.createElement(KpIcon, {
    shape: "circle",
    type: "icon-sousuo",
    onClick: onSearch
  }))), /*#__PURE__*/React.createElement(FileManageContext.Provider, {
    value: _objectSpread({
      settings: _objectSpread(_objectSpread({}, restSettings), {}, {
        defalutActive: active,
        sidebar: {
          width: width
        }
      }),
      currentSelectedList: currentSelectedList,
      searchValue: searchValue,
      beforeUpload: beforeUpload,
      onChangeState: onChangeState,
      dataSource: dataSource,
      onSearch: onSearch,
      onResetFetch: onResetFetch,
      activeIndex: activeIndex
    }, rest)
  }, /*#__PURE__*/React.createElement(ModalFilesContent, null)))));
};

export default BasisLayout;
