// @kuping/oto_component/lib/business/SearchInput/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireDefault(require("react"));
var _commonLib = require("@boluome/common-lib");
var _common = require("../../utils/common");
var _const = require("../../utils/const");
var _icon_arrow_down = _interopRequireDefault(require("./img/icon_arrow_down.png"));
var _icon_location = _interopRequireDefault(require("./img/icon_location.png"));
var _icon_search = _interopRequireDefault(require("./img/icon_search.png"));
require("./index.css");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var SearchInput = /*#__PURE__*/function (_React$Component) {
  _inheritsLoose(SearchInput, _React$Component);
  function SearchInput() {
    return _React$Component.apply(this, arguments) || this;
  }
  var _proto = SearchInput.prototype;
  _proto.handleClick = function handleClick(path) {
    var name = path === '/template/city' ? '城市选择' : '搜索';
    var _ref = this.props.data || {},
      basicAttribute = _ref.basicAttribute;
    var _ref2 = basicAttribute || {},
      monitor = _ref2.monitor;
    var _this$props = this.props,
      onSearchClick = _this$props.onSearchClick,
      componentName = _this$props.componentName;
    var config = _extends({
      name: name,
      path: path
    }, monitor || {});
    config.componentName = componentName || '定位';
    (0, _common.globalClick)(config, onSearchClick);
  };
  _proto.nativeSkip = function nativeSkip(config) {
    var componentName = this.props.componentName;
    var _ref3 = this.props.data || {},
      basicAttribute = _ref3.basicAttribute;
    var _ref4 = basicAttribute || {},
      monitor = _ref4.monitor;
    var row = _extends({}, config, monitor || {});
    row.componentName = componentName || '定位';
    (0, _common.globalClick)(row);
  };
  _proto.render = function render() {
    var _this = this;
    var _ref5 = this.props.data || {},
      _ref5$dataSource = _ref5.dataSource,
      dataSource = _ref5$dataSource === void 0 ? {} : _ref5$dataSource,
      basicAttribute = _ref5.basicAttribute;
    var _ref6 = basicAttribute || {},
      relationStyle = _ref6.relationStyle;
    var name = dataSource.name,
      imgUrl = dataSource.imgUrl;
    var cityName = (0, _commonLib.getStore)('selectCityPindao', 'session') ? (0, _commonLib.getStore)('selectCityPindao', 'session').cityName : _const.defaultCity;
    var Style2 = function Style2() {
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "location-search"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: "location-content",
        onClick: function onClick() {
          return _this.handleClick('/template/city');
        }
      }, cityName, /*#__PURE__*/_react["default"].createElement("img", {
        src: _icon_arrow_down["default"],
        alt: ""
      })), /*#__PURE__*/_react["default"].createElement("div", {
        className: "search-content",
        onClick: function onClick() {
          return _this.handleClick('/template/search');
        }
      }, /*#__PURE__*/_react["default"].createElement("img", {
        src: _icon_search["default"],
        alt: "",
        className: "am-icon"
      }), name));
    };
    var Style3 = function Style3() {
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "location-search-img"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: "location-search"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: "location-content",
        onClick: function onClick() {
          return _this.handleClick('/template/city');
        }
      }, cityName, /*#__PURE__*/_react["default"].createElement("img", {
        src: _icon_arrow_down["default"],
        alt: ""
      })), /*#__PURE__*/_react["default"].createElement("div", {
        className: "search-content",
        onClick: function onClick() {
          return _this.handleClick('/template/search');
        }
      }, /*#__PURE__*/_react["default"].createElement("img", {
        src: _icon_search["default"],
        alt: "",
        className: "am-icon"
      }), name)), /*#__PURE__*/_react["default"].createElement("img", {
        src: imgUrl,
        alt: "",
        className: "custom-img",
        onClick: function onClick() {
          return _this.nativeSkip(dataSource);
        }
      }));
    };
    var Style4 = function Style4() {
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "location-search-img"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: "location-search"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: "search-content",
        onClick: function onClick() {
          return _this.handleClick('/template/search');
        }
      }, /*#__PURE__*/_react["default"].createElement("img", {
        src: _icon_search["default"],
        alt: "",
        className: "am-icon"
      }), name)));
    };
    var StyleDefault = function StyleDefault() {
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "selected-location",
        onClick: function onClick() {
          return _this.handleClick('/template/city');
        }
      }, /*#__PURE__*/_react["default"].createElement("img", {
        src: _icon_location["default"],
        alt: ""
      }), /*#__PURE__*/_react["default"].createElement("span", {
        style: {
          marginLeft: '.08rem'
        }
      }, cityName), /*#__PURE__*/_react["default"].createElement("img", {
        src: "https://bosscdn.otosaas.com/test/20220830/20220830160343icon_right_arrow.png?path=0,518",
        alt: "",
        className: "am-icon"
      }));
    };
    var LocationEle = function LocationEle() {
      if (relationStyle === 2) {
        return /*#__PURE__*/_react["default"].createElement(Style2, null);
      }
      if (relationStyle === 3) {
        return /*#__PURE__*/_react["default"].createElement(Style3, null);
      }
      if (relationStyle === 4) {
        return /*#__PURE__*/_react["default"].createElement(Style4, null);
      }
      return /*#__PURE__*/_react["default"].createElement(StyleDefault, null);
    };
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: "template-location",
      style: {
        margin: relationStyle === 1 ? '0' : '0.2rem 0'
      }
    }, /*#__PURE__*/_react["default"].createElement(LocationEle, null));
  };
  return SearchInput;
}(_react["default"].Component);
SearchInput.defaultProps = {
  attr: {
    name: '定位'
  }
};
var _default = SearchInput;
exports["default"] = _default;
