// @kuping/oto_component/lib/types/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ActionSheetProps", {
  enumerable: true,
  get: function get() {
    return _actionSheet["default"];
  }
});
Object.defineProperty(exports, "AutoCenterProps", {
  enumerable: true,
  get: function get() {
    return _autoCenter["default"];
  }
});
Object.defineProperty(exports, "AvatarProps", {
  enumerable: true,
  get: function get() {
    return _avatar["default"];
  }
});
Object.defineProperty(exports, "BadgeProps", {
  enumerable: true,
  get: function get() {
    return _badge["default"];
  }
});
Object.defineProperty(exports, "ButtonProps", {
  enumerable: true,
  get: function get() {
    return _button["default"];
  }
});
Object.defineProperty(exports, "CapsuleTabsProps", {
  enumerable: true,
  get: function get() {
    return _capsuleTabs["default"];
  }
});
Object.defineProperty(exports, "CardProps", {
  enumerable: true,
  get: function get() {
    return _card["default"];
  }
});
Object.defineProperty(exports, "CascaderProps", {
  enumerable: true,
  get: function get() {
    return _cascadePicker["default"];
  }
});
Object.defineProperty(exports, "CascaderViewProps", {
  enumerable: true,
  get: function get() {
    return _cascaderView["default"];
  }
});
Object.defineProperty(exports, "CheckListProps", {
  enumerable: true,
  get: function get() {
    return _checkList["default"];
  }
});
Object.defineProperty(exports, "CheckboxProps", {
  enumerable: true,
  get: function get() {
    return _checkbox["default"];
  }
});
Object.defineProperty(exports, "CollapseProps", {
  enumerable: true,
  get: function get() {
    return _collapse["default"];
  }
});
Object.defineProperty(exports, "ConfigProviderProps", {
  enumerable: true,
  get: function get() {
    return _configProvider["default"];
  }
});
Object.defineProperty(exports, "DialogProps", {
  enumerable: true,
  get: function get() {
    return _dialog["default"];
  }
});
Object.defineProperty(exports, "DividerProps", {
  enumerable: true,
  get: function get() {
    return _divider["default"];
  }
});
Object.defineProperty(exports, "EllipsisProps", {
  enumerable: true,
  get: function get() {
    return _ellipsis["default"];
  }
});
Object.defineProperty(exports, "EmptyProps", {
  enumerable: true,
  get: function get() {
    return _empty["default"];
  }
});
Object.defineProperty(exports, "ErrorBlockProps", {
  enumerable: true,
  get: function get() {
    return _errorBlock["default"];
  }
});
Object.defineProperty(exports, "FloatingPanelProps", {
  enumerable: true,
  get: function get() {
    return _floatingPanel["default"];
  }
});
Object.defineProperty(exports, "FormProps", {
  enumerable: true,
  get: function get() {
    return _form["default"];
  }
});
Object.defineProperty(exports, "GridProps", {
  enumerable: true,
  get: function get() {
    return _grid["default"];
  }
});
Object.defineProperty(exports, "ImageProps", {
  enumerable: true,
  get: function get() {
    return _image["default"];
  }
});
Object.defineProperty(exports, "ImageViewerProps", {
  enumerable: true,
  get: function get() {
    return _imageViewer["default"];
  }
});
Object.defineProperty(exports, "IndexBarProps", {
  enumerable: true,
  get: function get() {
    return _indexBar["default"];
  }
});
Object.defineProperty(exports, "InfiniteScrollProps", {
  enumerable: true,
  get: function get() {
    return _infiniteScroll["default"];
  }
});
Object.defineProperty(exports, "InputProps", {
  enumerable: true,
  get: function get() {
    return _input["default"];
  }
});
Object.defineProperty(exports, "JumboTabsProps", {
  enumerable: true,
  get: function get() {
    return _jumboTabs["default"];
  }
});
Object.defineProperty(exports, "LoadingProps", {
  enumerable: true,
  get: function get() {
    return _loading["default"];
  }
});
Object.defineProperty(exports, "MaskProps", {
  enumerable: true,
  get: function get() {
    return _mask["default"];
  }
});
Object.defineProperty(exports, "ModalProps", {
  enumerable: true,
  get: function get() {
    return _modal["default"];
  }
});
Object.defineProperty(exports, "NavBarProps", {
  enumerable: true,
  get: function get() {
    return _navBar["default"];
  }
});
Object.defineProperty(exports, "PageIndicatorProps", {
  enumerable: true,
  get: function get() {
    return _pageIndicator["default"];
  }
});
Object.defineProperty(exports, "PickerProps", {
  enumerable: true,
  get: function get() {
    return _picker["default"];
  }
});
Object.defineProperty(exports, "PickerViewProps", {
  enumerable: true,
  get: function get() {
    return _pickerView["default"];
  }
});
Object.defineProperty(exports, "PopoverProps", {
  enumerable: true,
  get: function get() {
    return _popover["default"];
  }
});
Object.defineProperty(exports, "PopupProps", {
  enumerable: true,
  get: function get() {
    return _popup["default"];
  }
});
Object.defineProperty(exports, "ProgressBarProps", {
  enumerable: true,
  get: function get() {
    return _progressBar["default"];
  }
});
Object.defineProperty(exports, "ProgressCircleProps", {
  enumerable: true,
  get: function get() {
    return _progressCircle["default"];
  }
});
Object.defineProperty(exports, "PullToRefreshProps", {
  enumerable: true,
  get: function get() {
    return _pullToRefresh["default"];
  }
});
Object.defineProperty(exports, "RadioProps", {
  enumerable: true,
  get: function get() {
    return _radio["default"];
  }
});
Object.defineProperty(exports, "RateProps", {
  enumerable: true,
  get: function get() {
    return _rate["default"];
  }
});
Object.defineProperty(exports, "ResultProps", {
  enumerable: true,
  get: function get() {
    return _result["default"];
  }
});
Object.defineProperty(exports, "SafeAreaProps", {
  enumerable: true,
  get: function get() {
    return _safeArea["default"];
  }
});
Object.defineProperty(exports, "SearchBarProps", {
  enumerable: true,
  get: function get() {
    return _searchBar["default"];
  }
});
Object.defineProperty(exports, "SelectorProps", {
  enumerable: true,
  get: function get() {
    return _selector["default"];
  }
});
Object.defineProperty(exports, "SideBarProps", {
  enumerable: true,
  get: function get() {
    return _sideBar["default"];
  }
});
Object.defineProperty(exports, "SkeletonProps", {
  enumerable: true,
  get: function get() {
    return _skeleton["default"];
  }
});
Object.defineProperty(exports, "SliderProps", {
  enumerable: true,
  get: function get() {
    return _slider["default"];
  }
});
Object.defineProperty(exports, "SpaceProps", {
  enumerable: true,
  get: function get() {
    return _space["default"];
  }
});
Object.defineProperty(exports, "StepperProps", {
  enumerable: true,
  get: function get() {
    return _stepper["default"];
  }
});
Object.defineProperty(exports, "StepsProps", {
  enumerable: true,
  get: function get() {
    return _steps["default"];
  }
});
Object.defineProperty(exports, "SwipeActionProps", {
  enumerable: true,
  get: function get() {
    return _swipeAction["default"];
  }
});
Object.defineProperty(exports, "SwiperProps", {
  enumerable: true,
  get: function get() {
    return _swiper["default"];
  }
});
Object.defineProperty(exports, "SwitchProps", {
  enumerable: true,
  get: function get() {
    return _switch["default"];
  }
});
Object.defineProperty(exports, "TabBarProps", {
  enumerable: true,
  get: function get() {
    return _tabBar["default"];
  }
});
Object.defineProperty(exports, "TabsProps", {
  enumerable: true,
  get: function get() {
    return _tabs["default"];
  }
});
Object.defineProperty(exports, "TagProps", {
  enumerable: true,
  get: function get() {
    return _tag["default"];
  }
});
Object.defineProperty(exports, "TextAreaProps", {
  enumerable: true,
  get: function get() {
    return _textArea["default"];
  }
});
Object.defineProperty(exports, "ToastProps", {
  enumerable: true,
  get: function get() {
    return _toast["default"];
  }
});
Object.defineProperty(exports, "WaterMarkProps", {
  enumerable: true,
  get: function get() {
    return _waterMark["default"];
  }
});
var _actionSheet = _interopRequireDefault(require("antd-mobile/es/components/action-sheet"));
var _autoCenter = _interopRequireDefault(require("antd-mobile/es/components/auto-center"));
var _avatar = _interopRequireDefault(require("antd-mobile/es/components/avatar"));
var _badge = _interopRequireDefault(require("antd-mobile/es/components/badge"));
var _button = _interopRequireDefault(require("antd-mobile/es/components/button"));
var _capsuleTabs = _interopRequireDefault(require("antd-mobile/es/components/capsule-tabs"));
var _card = _interopRequireDefault(require("antd-mobile/es/components/card"));
var _cascadePicker = _interopRequireDefault(require("antd-mobile/es/components/cascade-picker"));
var _cascaderView = _interopRequireDefault(require("antd-mobile/es/components/cascader-view"));
var _checkList = _interopRequireDefault(require("antd-mobile/es/components/check-list"));
var _checkbox = _interopRequireDefault(require("antd-mobile/es/components/checkbox"));
var _collapse = _interopRequireDefault(require("antd-mobile/es/components/collapse"));
var _configProvider = _interopRequireDefault(require("antd-mobile/es/components/config-provider"));
var _dialog = _interopRequireDefault(require("antd-mobile/es/components/dialog"));
var _divider = _interopRequireDefault(require("antd-mobile/es/components/divider"));
var _ellipsis = _interopRequireDefault(require("antd-mobile/es/components/ellipsis"));
var _empty = _interopRequireDefault(require("antd-mobile/es/components/empty"));
var _errorBlock = _interopRequireDefault(require("antd-mobile/es/components/error-block"));
var _floatingPanel = _interopRequireDefault(require("antd-mobile/es/components/floating-panel"));
var _form = _interopRequireDefault(require("antd-mobile/es/components/form"));
var _grid = _interopRequireDefault(require("antd-mobile/es/components/grid"));
var _image = _interopRequireDefault(require("antd-mobile/es/components/image"));
var _imageViewer = _interopRequireDefault(require("antd-mobile/es/components/image-viewer"));
var _indexBar = _interopRequireDefault(require("antd-mobile/es/components/index-bar"));
var _infiniteScroll = _interopRequireDefault(require("antd-mobile/es/components/infinite-scroll"));
var _input = _interopRequireDefault(require("antd-mobile/es/components/input"));
var _jumboTabs = _interopRequireDefault(require("antd-mobile/es/components/jumbo-tabs"));
var _loading = _interopRequireDefault(require("antd-mobile/es/components/loading"));
var _mask = _interopRequireDefault(require("antd-mobile/es/components/mask"));
var _modal = _interopRequireDefault(require("antd-mobile/es/components/modal"));
var _navBar = _interopRequireDefault(require("antd-mobile/es/components/nav-bar"));
var _pageIndicator = _interopRequireDefault(require("antd-mobile/es/components/page-indicator"));
var _picker = _interopRequireDefault(require("antd-mobile/es/components/picker"));
var _pickerView = _interopRequireDefault(require("antd-mobile/es/components/picker-view"));
var _popover = _interopRequireDefault(require("antd-mobile/es/components/popover"));
var _popup = _interopRequireDefault(require("antd-mobile/es/components/popup"));
var _progressBar = _interopRequireDefault(require("antd-mobile/es/components/progress-bar"));
var _progressCircle = _interopRequireDefault(require("antd-mobile/es/components/progress-circle"));
var _pullToRefresh = _interopRequireDefault(require("antd-mobile/es/components/pull-to-refresh"));
var _radio = _interopRequireDefault(require("antd-mobile/es/components/radio"));
var _rate = _interopRequireDefault(require("antd-mobile/es/components/rate"));
var _result = _interopRequireDefault(require("antd-mobile/es/components/result"));
var _safeArea = _interopRequireDefault(require("antd-mobile/es/components/safe-area"));
var _searchBar = _interopRequireDefault(require("antd-mobile/es/components/search-bar"));
var _selector = _interopRequireDefault(require("antd-mobile/es/components/selector"));
var _sideBar = _interopRequireDefault(require("antd-mobile/es/components/side-bar"));
var _skeleton = _interopRequireDefault(require("antd-mobile/es/components/skeleton"));
var _slider = _interopRequireDefault(require("antd-mobile/es/components/slider"));
var _space = _interopRequireDefault(require("antd-mobile/es/components/space"));
var _stepper = _interopRequireDefault(require("antd-mobile/es/components/stepper"));
var _steps = _interopRequireDefault(require("antd-mobile/es/components/steps"));
var _swipeAction = _interopRequireDefault(require("antd-mobile/es/components/swipe-action"));
var _swiper = _interopRequireDefault(require("antd-mobile/es/components/swiper"));
var _switch = _interopRequireDefault(require("antd-mobile/es/components/switch"));
var _tabBar = _interopRequireDefault(require("antd-mobile/es/components/tab-bar"));
var _tabs = _interopRequireDefault(require("antd-mobile/es/components/tabs"));
var _tag = _interopRequireDefault(require("antd-mobile/es/components/tag"));
var _textArea = _interopRequireDefault(require("antd-mobile/es/components/text-area"));
var _toast = _interopRequireDefault(require("antd-mobile/es/components/toast"));
var _waterMark = _interopRequireDefault(require("antd-mobile/es/components/water-mark"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
