// @kuping/oto_component/lib/tools/HotArea/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireDefault(require("react"));
var _common = require("../../utils/common");
require("./index.css");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var SOURCE_WIDTH = 489;
var ratio = window.innerWidth / SOURCE_WIDTH;
var HotArea = function HotArea(_ref) {
  var _data$dataSource;
  var data = _ref.data;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "drag-widget-hot-area"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "drag-widget-hot-area-image"
  }, /*#__PURE__*/_react["default"].createElement("img", {
    src: data.basicAttribute.image || 'https://bosscdn.otosaas.com/test/20221108/20221108160310%E7%83%AD%E5%8C%BA.png?path=0,650,665',
    alt: "\u70ED\u533A\u56FE"
  })), (_data$dataSource = data.dataSource) === null || _data$dataSource === void 0 ? void 0 : _data$dataSource.map(function (item, index) {
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: "drag-widget-hot-area-item",
      key: index,
      style: {
        width: item.width * ratio,
        height: item.height * ratio,
        left: item.left * ratio,
        top: item.top * ratio
      },
      onClick: function onClick() {
        return (0, _common.globalClick)(_extends({
          name: item.link,
          pathUrl: item.link
        }, data.basicAttribute.monitor));
      }
    });
  }));
};
HotArea.displayName = 'HotArea';
HotArea.defaultProps = {
  attr: {
    name: '热区'
  }
};
var _default = HotArea;
exports["default"] = _default;
