// @kp-react-visual-design/design/lib/components/Drag/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("./index.styled");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Drag = function Drag(_ref) {
  var initialList = _ref.initialList,
      render = _ref.render,
      onDragStart = _ref.onDragStart,
      onDragEnd = _ref.onDragEnd;

  var handleDragStart = function handleDragStart(widgetName, e) {
    e.dataTransfer.dropEffect = 'move';
    e.dataTransfer.setData('widgetName', widgetName);
    onDragStart === null || onDragStart === void 0 ? void 0 : onDragStart(e);
  };

  var handleDragEnd = function handleDragEnd(e) {
    e.dataTransfer.effectAllowed = 'none';
    e.dataTransfer.clearData();
    onDragEnd === null || onDragEnd === void 0 ? void 0 : onDragEnd(e);
  };

  return /*#__PURE__*/_react.default.createElement(_index.Conatainer, null, initialList.map(function (item, index) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "drag-box",
      draggable: true,
      key: index,
      onDragStart: function onDragStart(e) {
        return handleDragStart(item.en, e);
      },
      onDragEnd: function onDragEnd(e) {
        return handleDragEnd(e);
      }
    }, render({
      item: item,
      index: index
    }));
  }));
};

var _default = Drag;
exports.default = _default;
