// @kp-react-visual-design/design/lib/VisualDesign/CompositePanel/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _antd = require("antd");

var _context = _interopRequireDefault(require("../context"));

var _components = require("../../components");

var _index = require("./index.styled");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var CompositePanel = function CompositePanel() {
  var _useContext = (0, _react.useContext)(_context.default),
      onChangeState = _useContext.onChangeState,
      initialWidgets = _useContext.initialWidgets,
      isOnLoad = _useContext.isOnLoad,
      loading = _useContext.loading;

  var businessWidgets = (0, _react.useMemo)(function () {
    return initialWidgets.filter(function (item) {
      return item.type === 'business';
    });
  }, [initialWidgets.length]);
  var sceneWidgets = (0, _react.useMemo)(function () {
    return initialWidgets.filter(function (item) {
      return item.type === 'scene';
    });
  }, [initialWidgets.length]);
  var functionsWidgets = (0, _react.useMemo)(function () {
    return initialWidgets.filter(function (item) {
      return item.type === 'functions';
    });
  }, [initialWidgets.length]);
  return /*#__PURE__*/_react.default.createElement(_index.Wrapper, null, /*#__PURE__*/_react.default.createElement(_index.CollapseStyled, {
    defaultActiveKey: ['1', '2', '3']
  }, /*#__PURE__*/_react.default.createElement(_index.PanelStyled, {
    header: "\u4E1A\u52A1\u7C7B",
    key: "1"
  }, !!businessWidgets.length ? /*#__PURE__*/_react.default.createElement(_components.Drag, {
    initialList: businessWidgets,
    render: function render(_ref) {
      var item = _ref.item;
      return item.visible ? /*#__PURE__*/_react.default.createElement(_index.WidgetItemStyled, null, /*#__PURE__*/_react.default.createElement("img", {
        className: "icon",
        src: item.icon,
        alt: ""
      }), /*#__PURE__*/_react.default.createElement("div", {
        className: "name"
      }, item.name)) : null;
    },
    onDragStart: function onDragStart() {
      return isOnLoad && !loading && onChangeState({
        isDragging: true
      });
    },
    onDragEnd: function onDragEnd() {
      return isOnLoad && !loading && onChangeState({
        isDragging: false
      });
    }
  }) : /*#__PURE__*/_react.default.createElement(_antd.Empty, {
    description: "\u6682\u65E0\u6570\u636E"
  })), /*#__PURE__*/_react.default.createElement(_index.PanelStyled, {
    header: "\u573A\u666F\u7C7B",
    key: "2"
  }, !!sceneWidgets.length ? /*#__PURE__*/_react.default.createElement(_components.Drag, {
    initialList: sceneWidgets,
    render: function render(_ref2) {
      var item = _ref2.item;
      return /*#__PURE__*/_react.default.createElement(_index.WidgetItemStyled, null, /*#__PURE__*/_react.default.createElement("img", {
        className: "icon",
        src: item.icon,
        alt: ""
      }), /*#__PURE__*/_react.default.createElement("div", {
        className: "name"
      }, item.name));
    },
    onDragStart: function onDragStart() {
      return isOnLoad && !loading && onChangeState({
        isDragging: true
      });
    },
    onDragEnd: function onDragEnd() {
      return isOnLoad && !loading && onChangeState({
        isDragging: false
      });
    }
  }) : /*#__PURE__*/_react.default.createElement(_antd.Empty, {
    description: "\u6682\u65E0\u6570\u636E"
  })), /*#__PURE__*/_react.default.createElement(_index.PanelStyled, {
    header: "\u529F\u80FD\u7C7B",
    key: "3"
  }, !!functionsWidgets.length ? /*#__PURE__*/_react.default.createElement(_components.Drag, {
    initialList: functionsWidgets,
    render: function render(_ref3) {
      var item = _ref3.item;
      return /*#__PURE__*/_react.default.createElement(_index.WidgetItemStyled, null, /*#__PURE__*/_react.default.createElement("img", {
        className: "icon",
        src: item.icon,
        alt: ""
      }), /*#__PURE__*/_react.default.createElement("div", {
        className: "name"
      }, item.name));
    },
    onDragStart: function onDragStart() {
      return isOnLoad && !loading && onChangeState({
        isDragging: true
      });
    },
    onDragEnd: function onDragEnd() {
      return isOnLoad && !loading && onChangeState({
        isDragging: false
      });
    }
  }) : /*#__PURE__*/_react.default.createElement(_antd.Empty, {
    description: "\u6682\u65E0\u6570\u636E"
  }))));
};

var _default = CompositePanel;
exports.default = _default;
