// @kuping/oto_component/lib/business/CategoryRecommendation/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireDefault(require("react"));
var _common = require("../../utils/common");
var _default2 = require("./default");
var _CustomImage = _interopRequireDefault(require("../CustomImage"));
require("./index.css");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var CategoryRecommendation = /*#__PURE__*/function (_React$Component) {
  _inheritsLoose(CategoryRecommendation, _React$Component);
  function CategoryRecommendation() {
    return _React$Component.apply(this, arguments) || this;
  }
  var _proto = CategoryRecommendation.prototype;
  _proto.handleClick = function handleClick(config) {
    var onClick = this.props.onClick;
    config.componentName = '分类推荐';
    (0, _common.globalClick)(config, onClick);
  };
  _proto.render = function render() {
    var _this = this;
    var data = this.props.data;
    var _ref = data || {},
      _dataSource = _ref.dataSource,
      basicAttribute = _ref.basicAttribute;
    var _ref2 = basicAttribute || {},
      title = _ref2.title,
      _ref2$monitor = _ref2.monitor,
      monitor = _ref2$monitor === void 0 ? {} : _ref2$monitor;
    var dataSource = _dataSource && _dataSource.length ? _dataSource : _default2.defaultDataSource;
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: "template-category-wrap"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "category-title"
    }, title), /*#__PURE__*/_react["default"].createElement("div", {
      className: "category-list"
    }, dataSource.map(function (v, index) {
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "category-item",
        key: index,
        onClick: function onClick() {
          return _this.handleClick(_extends({}, v, monitor));
        }
      }, /*#__PURE__*/_react["default"].createElement(_CustomImage["default"], {
        src: v.imgUrl,
        alt: "",
        className: "category-img"
      }), /*#__PURE__*/_react["default"].createElement("div", {
        className: "category-name"
      }, v.name));
    })));
  };
  return CategoryRecommendation;
}(_react["default"].Component);
CategoryRecommendation.defaultProps = {
  attr: {
    name: '分类推荐'
  }
};
var _default = CategoryRecommendation;
exports["default"] = _default;
