// @kuping/oto_component/lib/business/Items/Price/Price4.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var Price4 = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Price4, _Component);
  function Price4() {
    return _Component.apply(this, arguments) || this;
  }
  var _proto = Price4.prototype;
  _proto.render = function render() {
    var _this$props = this.props,
      price = _this$props.price,
      originPrice = _this$props.originPrice,
      _this$props$saleTipTe = _this$props.saleTipText,
      saleTipText = _this$props$saleTipTe === void 0 ? '' : _this$props$saleTipTe;
    var linePrice = price < originPrice ? originPrice : price;
    var intPrice = price.toString().split('.')[0];
    var floatPrice = price.toString().split('.')[1];
    var EstimatePriceEl = function EstimatePriceEl() {
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "special-price"
      }, /*#__PURE__*/_react["default"].createElement("span", {
        style: {
          marginRight: '0.05rem'
        }
      }, saleTipText, ":"), "\xA5", /*#__PURE__*/_react["default"].createElement("span", {
        className: "price-s"
      }, intPrice, floatPrice > 0 && /*#__PURE__*/_react["default"].createElement("span", null, ".", floatPrice)));
    };
    var OriginPriceEl = function OriginPriceEl() {
      return linePrice > price && /*#__PURE__*/_react["default"].createElement("div", {
        className: "price-origin"
      }, /*#__PURE__*/_react["default"].createElement("s", null, "\xA5", linePrice));
    };
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: "price-wrap price-style-4"
    }, /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(EstimatePriceEl, null), /*#__PURE__*/_react["default"].createElement(OriginPriceEl, null)), /*#__PURE__*/_react["default"].createElement("div", {
      className: "price-button"
    }, "\u62A2\u8D2D"));
  };
  return Price4;
}(_react.Component);
Price4.defaultProps = {
  price: 0,
  originPrice: 0
};
var _default = Price4;
exports["default"] = _default;
