// @kuping/oto_component/lib/business/Items/Price/Price3.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var Price3 = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Price3, _Component);
  function Price3() {
    return _Component.apply(this, arguments) || this;
  }
  var _proto = Price3.prototype;
  _proto.render = function render() {
    var _this$props = this.props,
      price = _this$props.price,
      originPrice = _this$props.originPrice,
      saleTipText = _this$props.saleTipText,
      newLine = _this$props.newLine;
    var currentPrice = price < originPrice ? parseFloat(originPrice) : price;
    var intPrice = currentPrice.toString().split('.')[0];
    var floatPrice = currentPrice.toString().split('.')[1];
    console.log('newLine', newLine);
    var EstimatePriceEl = function EstimatePriceEl(_ref) {
      var isMargin = _ref.isMargin;
      return price < originPrice && /*#__PURE__*/_react["default"].createElement("span", {
        className: "estimatedPrice",
        style: {
          marginLeft: isMargin ? '0' : '0.14rem'
        }
      }, /*#__PURE__*/_react["default"].createElement("img", {
        src: 'https://bosscdn.otosaas.com/test/20220726/20220726181757triangle.svg?path=0,391',
        className: "icon-triangle"
      }), saleTipText, ":", price);
    };
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: "price3-wrap"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      className: "info-price commodity-info",
      style: {
        display: 'flex',
        flexDirection: 'row'
      }
    }, /*#__PURE__*/_react["default"].createElement("span", {
      className: "price-current"
    }, /*#__PURE__*/_react["default"].createElement("span", null, "\xA5"), /*#__PURE__*/_react["default"].createElement("b", null, intPrice, floatPrice && /*#__PURE__*/_react["default"].createElement("span", null, ".", floatPrice))), intPrice <= 999.99 && !newLine && /*#__PURE__*/_react["default"].createElement(EstimatePriceEl, null)), (intPrice > 999.99 || newLine) && /*#__PURE__*/_react["default"].createElement(EstimatePriceEl, {
      isMargin: true
    }));
  };
  return Price3;
}(_react.Component);
Price3.defaultProps = {
  originPrice: 0,
  price: 0,
  newLine: false
};
var _default = Price3;
exports["default"] = _default;
