// @kuping/oto_component/lib/business/Marketing/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var _common = require("../../utils/common");
var _default2 = require("./default");
var _CustomImage = _interopRequireDefault(require("../CustomImage"));
require("./index.css");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var Marketing = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Marketing, _Component);
  function Marketing() {
    return _Component.apply(this, arguments) || this;
  }
  var _proto = Marketing.prototype;
  _proto.handleClick = function handleClick(config) {
    var _ref = this.props.data || {},
      _ref$basicAttribute = _ref.basicAttribute,
      basicAttribute = _ref$basicAttribute === void 0 ? {} : _ref$basicAttribute;
    var onClick = this.props.onClick;
    var row = _extends({}, config, basicAttribute.monitor || {});
    row.componentName = '营销位';
    (0, _common.globalClick)(row, onClick);
  };
  _proto.render = function render() {
    var _this = this;
    var _this$props$data = this.props.data,
      _dataSource = _this$props$data.dataSource,
      basicAttribute = _this$props$data.basicAttribute;
    var _ref2 = basicAttribute || {},
      _ref2$relationStyle = _ref2.relationStyle,
      relationStyle = _ref2$relationStyle === void 0 ? 3 : _ref2$relationStyle;
    var number = {
      1: 2,
      2: 3,
      3: 3,
      4: 4,
      5: 4
    };
    var dataSource = _dataSource && _dataSource.length ? _dataSource : _default2.defaultDataSource.slice(0, number[relationStyle]);
    var stylesObj = {
      2: {
        gridTemplateColumns: 'repeat(3, 1fr)',
        gridTemplateRows: 'repeat(1, minmax(0, 1fr))'
      },
      4: {
        gridTemplateColumns: 'repeat(4, 1fr)',
        gridTemplateRows: 'repeat(1, minmax(0, 1fr))'
      },
      "default": {
        gridTemplateColumns: 'repeat(12, 1fr)',
        gridTemplateRows: 'repeat(2, minmax(0, 1fr))'
      }
    };
    var ShowcaseWrap = function ShowcaseWrap(props) {
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "template-grid",
        style: stylesObj[relationStyle] ? stylesObj[relationStyle] : stylesObj["default"]
      }, props.children);
    };
    var ShowcaseItem = function ShowcaseItem(props) {
      var item = props.item;
      return /*#__PURE__*/_react["default"].createElement("div", {
        onClick: function onClick() {
          return _this.handleClick(item);
        },
        className: "showcase"
      }, props.children);
    };
    // 默认 左一右二
    var ShowcaseDefaultEl = function ShowcaseDefaultEl() {
      return /*#__PURE__*/_react["default"].createElement(ShowcaseWrap, null, dataSource.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement("div", {
          key: index,
          className: index === 0 ? 'grid-item-row' : 'grid-item'
        }, /*#__PURE__*/_react["default"].createElement(ShowcaseItem, {
          item: item
        }, /*#__PURE__*/_react["default"].createElement(_CustomImage["default"], {
          src: item.imgUrl,
          alt: "",
          className: index === 0 ? 'h-350' : 'h-167'
        })));
      }));
    };
    // 一行三个
    var ShowcaseAequilateEl = function ShowcaseAequilateEl() {
      return /*#__PURE__*/_react["default"].createElement(ShowcaseWrap, null, dataSource.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement(ShowcaseItem, {
          item: item,
          key: index
        }, /*#__PURE__*/_react["default"].createElement(_CustomImage["default"], {
          src: item.imgUrl,
          alt: "",
          style: {
            height: 'auto'
          }
        }));
      }));
    };
    // 一行两个
    var ShowcaseTwoEl = function ShowcaseTwoEl() {
      return /*#__PURE__*/_react["default"].createElement(ShowcaseWrap, null, dataSource.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement("div", {
          key: index,
          className: 'grid-item-row'
        }, /*#__PURE__*/_react["default"].createElement(ShowcaseItem, {
          item: item
        }, /*#__PURE__*/_react["default"].createElement(_CustomImage["default"], {
          src: item.imgUrl,
          className: 'h-167'
        })));
      }));
    };
    // 一行四个
    var ShowcaseFourEl = function ShowcaseFourEl() {
      return /*#__PURE__*/_react["default"].createElement(ShowcaseWrap, null, dataSource.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement(ShowcaseItem, {
          item: item,
          key: index
        }, /*#__PURE__*/_react["default"].createElement(_CustomImage["default"], {
          src: item.imgUrl,
          alt: "",
          style: {
            height: '1.24rem'
          }
        }));
      }));
    };
    // 左一右三
    var ShowcaseFiveEl = function ShowcaseFiveEl() {
      return /*#__PURE__*/_react["default"].createElement(ShowcaseWrap, null, dataSource.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement("div", {
          key: index,
          className: 'grid-item',
          style: {
            gridArea: index === 0 ? 'span 3 / span 6 / span 3 / span 6' : 'span 1 / span 6'
          }
        }, /*#__PURE__*/_react["default"].createElement(ShowcaseItem, {
          item: item
        }, /*#__PURE__*/_react["default"].createElement(_CustomImage["default"], {
          src: item.imgUrl,
          alt: "",
          className: index === 0 ? 'h-533' : 'h-167'
        })));
      }));
    };
    var showcaseArr = {
      1: /*#__PURE__*/_react["default"].createElement(ShowcaseTwoEl, null),
      2: /*#__PURE__*/_react["default"].createElement(ShowcaseAequilateEl, null),
      3: /*#__PURE__*/_react["default"].createElement(ShowcaseDefaultEl, null),
      4: /*#__PURE__*/_react["default"].createElement(ShowcaseFourEl, null),
      5: /*#__PURE__*/_react["default"].createElement(ShowcaseFiveEl, null)
    };
    return showcaseArr[relationStyle];
  };
  return Marketing;
}(_react.Component);
Marketing.defaultProps = {
  data: {
    dataSource: [],
    basicAttribute: {
      relationStyle: 3
    }
  },
  attr: {
    name: '营销位'
  }
};
var _default = Marketing;
exports["default"] = _default;
