// @kp-react-visual-design/design/lib/components/TableSearch/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _antd = require("antd");

var _index = require("./index.styled");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var components = {
  Input: _antd.Input,
  Select: _antd.Select,
  DatePicker: _antd.DatePicker,
  RangePicker: _antd.DatePicker.RangePicker
};

var SearchInputs = function SearchInputs(_ref) {
  var options = _ref.options,
      loading = _ref.loading,
      onChange = _ref.onChange,
      onSearch = _ref.onSearch,
      onReset = _ref.onReset,
      values = _ref.values;

  var handleChange = function handleChange(name, componentName, e) {
    if (componentName === 'Input') {
      onChange === null || onChange === void 0 ? void 0 : onChange(_defineProperty({}, name, e.target.value));
    } else {
      onChange === null || onChange === void 0 ? void 0 : onChange(_defineProperty({}, name, e));
    }
  };

  return /*#__PURE__*/_react.default.createElement(_index.Container, null, /*#__PURE__*/_react.default.createElement(_antd.Row, null, options.map(function (item, index) {
    var Component = components[item.componentName];
    return /*#__PURE__*/_react.default.createElement(_index.StyledCol, {
      span: 8,
      key: index
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: "label"
    }, item.label, ":"), /*#__PURE__*/_react.default.createElement(Component, {
      defaultValue: item.defaultValue,
      placeholder: item.placeholder,
      onChange: function onChange(e) {
        return handleChange(item.name, item.componentName, e);
      },
      style: {
        flex: '1 0 0'
      },
      options: item.options,
      value: item.componentName === 'RangePicker' ? values.date : values[item.name]
    }));
  })), !!options.length && /*#__PURE__*/_react.default.createElement(_index.ButtonContainer, null, /*#__PURE__*/_react.default.createElement(_antd.Button, {
    onClick: onReset,
    style: {
      marginRight: 10
    }
  }, "\u91CD\u7F6E"), /*#__PURE__*/_react.default.createElement(_antd.Button, {
    type: "primary",
    loading: loading,
    disabled: loading,
    onClick: onSearch
  }, "\u67E5\u8BE2")));
};

SearchInputs.defaultProps = {
  options: [],
  loading: false,
  onChange: function onChange() {}
};
var _default = SearchInputs;
exports.default = _default;
