// @kp-react-visual-design/design/lib/VisualDesign/ViewportPanel/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _antd = require("antd");

var _icons = require("@ant-design/icons");

var _qrcode = require("qrcode.react");

var _context = _interopRequireDefault(require("../context"));

var _index = require("./index.styled");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var environment = process.env.NODE_ENV === 'development' || location.host.includes('test') ? 'dev-' : '';

var ViewportPanel = function ViewportPanel(_ref) {
  var children = _ref.children;

  var _useContext = (0, _react.useContext)(_context.default),
      onChangeState = _useContext.onChangeState,
      isPreviewing = _useContext.isPreviewing,
      record = _useContext.record,
      appCode = _useContext.appCode,
      selectedList = _useContext.selectedList; // const handleRefresh = _.debounce(() => onChangeState({ refresh: new Date().getTime() }), 200);


  var city = (0, _react.useMemo)(function () {
    var _HeaderConfig$data, _HeaderConfig$data$da, _HeaderConfig$data$da2;

    var HeaderConfig = selectedList.find(function (item) {
      return item.name === 'Header';
    });
    return encodeURIComponent((HeaderConfig === null || HeaderConfig === void 0 ? void 0 : (_HeaderConfig$data = HeaderConfig.data) === null || _HeaderConfig$data === void 0 ? void 0 : (_HeaderConfig$data$da = _HeaderConfig$data.dataSource.city) === null || _HeaderConfig$data$da === void 0 ? void 0 : (_HeaderConfig$data$da2 = _HeaderConfig$data$da[0]) === null || _HeaderConfig$data$da2 === void 0 ? void 0 : _HeaderConfig$data$da2[1]) || '');
  }, [JSON.stringify(selectedList)]);
  var link = (0, _react.useMemo)(function () {
    var polymerizeCode = record.polymerizeCode,
        parentPolymerizeCode = record.parentPolymerizeCode,
        polymerizeType = record.polymerizeType;
    var channelID = polymerizeType === 1 ? polymerizeCode : parentPolymerizeCode;
    return "https://".concat(environment).concat(appCode, ".otosaas.com/template/").concat(channelID);
  }, [JSON.stringify(record)]);

  var handleClickPreview = function handleClickPreview() {
    onChangeState({
      isPreviewing: !isPreviewing
    });

    if (!isPreviewing) {
      _antd.message.info('建议使用手机扫码预览');
    }
  };

  return /*#__PURE__*/_react.default.createElement(_index.Wrapper, null, /*#__PURE__*/_react.default.createElement(_index.ControlsBarStyled, null, /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_antd.Button.Group, {
    size: "small"
  }, /*#__PURE__*/_react.default.createElement(_antd.Popover, {
    content: /*#__PURE__*/_react.default.createElement(_qrcode.QRCodeSVG, {
      value: "".concat(link, "?city=").concat(city, "&preview=1")
    }),
    placement: "leftTop"
  }, /*#__PURE__*/_react.default.createElement(_antd.Tooltip, {
    title: "\u626B\u7801\u9884\u89C8"
  }, /*#__PURE__*/_react.default.createElement(_antd.Button, {
    size: "small",
    icon: /*#__PURE__*/_react.default.createElement(_icons.QrcodeOutlined, null)
  }))), /*#__PURE__*/_react.default.createElement(_antd.Tooltip, {
    title: "\u5728\u7EBF\u9884\u89C8"
  }, /*#__PURE__*/_react.default.createElement(_antd.Button, {
    size: "small",
    onClick: handleClickPreview,
    icon: isPreviewing ? /*#__PURE__*/_react.default.createElement(_icons.FormatPainterOutlined, null) : /*#__PURE__*/_react.default.createElement(_icons.EyeOutlined, null)
  }))))), /*#__PURE__*/_react.default.createElement(_index.Container, null, children));
};

var _default = ViewportPanel;
exports.default = _default;
