// @kp-react-visual-design/design/lib/VisualDesign/SettingsPanel/schema-field.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@formily/react");

var _icons = require("@ant-design/icons");

var _antd = require("antd");

var _antd2 = require("@formily/antd");

var _components = require("../../components");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SchemaField = (0, _react2.createSchemaField)({
  components: {
    Radio: _antd2.Radio,
    Checkbox: _antd2.Checkbox,
    Input: _antd2.Input,
    DateRangePicker: _antd2.DatePicker,
    YearPicker: _antd2.DatePicker,
    TimeRangePicker: _antd2.TimePicker,
    FormItem: _antd2.FormItem,
    Password: _antd2.Password,
    ArrayCards: _antd2.ArrayCards,
    ArrayItems: _antd2.ArrayItems,
    Select: _antd2.Select,
    HotArea: _components.HotArea,
    Space: _antd2.Space,
    Card: _antd.Card,
    NumberPicker: _antd2.NumberPicker,
    Cascader: _antd2.Cascader,
    FileManage: _components.FileManage,
    LinksButton: _components.LinksButton,
    Switch: _antd2.Switch,
    LinkCoupon: _components.LinkCoupon,
    LinkGoods: _components.LinkGoods,
    LinkRecommendId: _components.LinkRecommendId
  },
  scope: {
    icon: function icon() {
      return /*#__PURE__*/_react.default.createElement(_icons.LoadingOutlined, null);
    }
  }
});
var _default = SchemaField;
exports.default = _default;
