// @kuping/oto_component/lib/business/MasonryLayouts/Filter/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var _antdMobileV = require("antd-mobile-v5");
var _selected = _interopRequireDefault(require("./img/selected.png"));
require("./index.css");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _regeneratorRuntime() { "use strict"; /*! regenerator-runtime -- Copyright (c) 2014-present, Facebook, Inc. -- license (MIT): https://github.com/facebook/regenerator/blob/main/LICENSE */ _regeneratorRuntime = function _regeneratorRuntime() { return exports; }; var exports = {}, Op = Object.prototype, hasOwn = Op.hasOwnProperty, defineProperty = Object.defineProperty || function (obj, key, desc) { obj[key] = desc.value; }, $Symbol = "function" == typeof Symbol ? Symbol : {}, iteratorSymbol = $Symbol.iterator || "@@iterator", asyncIteratorSymbol = $Symbol.asyncIterator || "@@asyncIterator", toStringTagSymbol = $Symbol.toStringTag || "@@toStringTag"; function define(obj, key, value) { return Object.defineProperty(obj, key, { value: value, enumerable: !0, configurable: !0, writable: !0 }), obj[key]; } try { define({}, ""); } catch (err) { define = function define(obj, key, value) { return obj[key] = value; }; } function wrap(innerFn, outerFn, self, tryLocsList) { var protoGenerator = outerFn && outerFn.prototype instanceof Generator ? outerFn : Generator, generator = Object.create(protoGenerator.prototype), context = new Context(tryLocsList || []); return defineProperty(generator, "_invoke", { value: makeInvokeMethod(innerFn, self, context) }), generator; } function tryCatch(fn, obj, arg) { try { return { type: "normal", arg: fn.call(obj, arg) }; } catch (err) { return { type: "throw", arg: err }; } } exports.wrap = wrap; var ContinueSentinel = {}; function Generator() {} function GeneratorFunction() {} function GeneratorFunctionPrototype() {} var IteratorPrototype = {}; define(IteratorPrototype, iteratorSymbol, function () { return this; }); var getProto = Object.getPrototypeOf, NativeIteratorPrototype = getProto && getProto(getProto(values([]))); NativeIteratorPrototype && NativeIteratorPrototype !== Op && hasOwn.call(NativeIteratorPrototype, iteratorSymbol) && (IteratorPrototype = NativeIteratorPrototype); var Gp = GeneratorFunctionPrototype.prototype = Generator.prototype = Object.create(IteratorPrototype); function defineIteratorMethods(prototype) { ["next", "throw", "return"].forEach(function (method) { define(prototype, method, function (arg) { return this._invoke(method, arg); }); }); } function AsyncIterator(generator, PromiseImpl) { function invoke(method, arg, resolve, reject) { var record = tryCatch(generator[method], generator, arg); if ("throw" !== record.type) { var result = record.arg, value = result.value; return value && "object" == _typeof(value) && hasOwn.call(value, "__await") ? PromiseImpl.resolve(value.__await).then(function (value) { invoke("next", value, resolve, reject); }, function (err) { invoke("throw", err, resolve, reject); }) : PromiseImpl.resolve(value).then(function (unwrapped) { result.value = unwrapped, resolve(result); }, function (error) { return invoke("throw", error, resolve, reject); }); } reject(record.arg); } var previousPromise; defineProperty(this, "_invoke", { value: function value(method, arg) { function callInvokeWithMethodAndArg() { return new PromiseImpl(function (resolve, reject) { invoke(method, arg, resolve, reject); }); } return previousPromise = previousPromise ? previousPromise.then(callInvokeWithMethodAndArg, callInvokeWithMethodAndArg) : callInvokeWithMethodAndArg(); } }); } function makeInvokeMethod(innerFn, self, context) { var state = "suspendedStart"; return function (method, arg) { if ("executing" === state) throw new Error("Generator is already running"); if ("completed" === state) { if ("throw" === method) throw arg; return doneResult(); } for (context.method = method, context.arg = arg;;) { var delegate = context.delegate; if (delegate) { var delegateResult = maybeInvokeDelegate(delegate, context); if (delegateResult) { if (delegateResult === ContinueSentinel) continue; return delegateResult; } } if ("next" === context.method) context.sent = context._sent = context.arg;else if ("throw" === context.method) { if ("suspendedStart" === state) throw state = "completed", context.arg; context.dispatchException(context.arg); } else "return" === context.method && context.abrupt("return", context.arg); state = "executing"; var record = tryCatch(innerFn, self, context); if ("normal" === record.type) { if (state = context.done ? "completed" : "suspendedYield", record.arg === ContinueSentinel) continue; return { value: record.arg, done: context.done }; } "throw" === record.type && (state = "completed", context.method = "throw", context.arg = record.arg); } }; } function maybeInvokeDelegate(delegate, context) { var method = delegate.iterator[context.method]; if (undefined === method) { if (context.delegate = null, "throw" === context.method) { if (delegate.iterator["return"] && (context.method = "return", context.arg = undefined, maybeInvokeDelegate(delegate, context), "throw" === context.method)) return ContinueSentinel; context.method = "throw", context.arg = new TypeError("The iterator does not provide a 'throw' method"); } return ContinueSentinel; } var record = tryCatch(method, delegate.iterator, context.arg); if ("throw" === record.type) return context.method = "throw", context.arg = record.arg, context.delegate = null, ContinueSentinel; var info = record.arg; return info ? info.done ? (context[delegate.resultName] = info.value, context.next = delegate.nextLoc, "return" !== context.method && (context.method = "next", context.arg = undefined), context.delegate = null, ContinueSentinel) : info : (context.method = "throw", context.arg = new TypeError("iterator result is not an object"), context.delegate = null, ContinueSentinel); } function pushTryEntry(locs) { var entry = { tryLoc: locs[0] }; 1 in locs && (entry.catchLoc = locs[1]), 2 in locs && (entry.finallyLoc = locs[2], entry.afterLoc = locs[3]), this.tryEntries.push(entry); } function resetTryEntry(entry) { var record = entry.completion || {}; record.type = "normal", delete record.arg, entry.completion = record; } function Context(tryLocsList) { this.tryEntries = [{ tryLoc: "root" }], tryLocsList.forEach(pushTryEntry, this), this.reset(!0); } function values(iterable) { if (iterable) { var iteratorMethod = iterable[iteratorSymbol]; if (iteratorMethod) return iteratorMethod.call(iterable); if ("function" == typeof iterable.next) return iterable; if (!isNaN(iterable.length)) { var i = -1, next = function next() { for (; ++i < iterable.length;) { if (hasOwn.call(iterable, i)) return next.value = iterable[i], next.done = !1, next; } return next.value = undefined, next.done = !0, next; }; return next.next = next; } } return { next: doneResult }; } function doneResult() { return { value: undefined, done: !0 }; } return GeneratorFunction.prototype = GeneratorFunctionPrototype, defineProperty(Gp, "constructor", { value: GeneratorFunctionPrototype, configurable: !0 }), defineProperty(GeneratorFunctionPrototype, "constructor", { value: GeneratorFunction, configurable: !0 }), GeneratorFunction.displayName = define(GeneratorFunctionPrototype, toStringTagSymbol, "GeneratorFunction"), exports.isGeneratorFunction = function (genFun) { var ctor = "function" == typeof genFun && genFun.constructor; return !!ctor && (ctor === GeneratorFunction || "GeneratorFunction" === (ctor.displayName || ctor.name)); }, exports.mark = function (genFun) { return Object.setPrototypeOf ? Object.setPrototypeOf(genFun, GeneratorFunctionPrototype) : (genFun.__proto__ = GeneratorFunctionPrototype, define(genFun, toStringTagSymbol, "GeneratorFunction")), genFun.prototype = Object.create(Gp), genFun; }, exports.awrap = function (arg) { return { __await: arg }; }, defineIteratorMethods(AsyncIterator.prototype), define(AsyncIterator.prototype, asyncIteratorSymbol, function () { return this; }), exports.AsyncIterator = AsyncIterator, exports.async = function (innerFn, outerFn, self, tryLocsList, PromiseImpl) { void 0 === PromiseImpl && (PromiseImpl = Promise); var iter = new AsyncIterator(wrap(innerFn, outerFn, self, tryLocsList), PromiseImpl); return exports.isGeneratorFunction(outerFn) ? iter : iter.next().then(function (result) { return result.done ? result.value : iter.next(); }); }, defineIteratorMethods(Gp), define(Gp, toStringTagSymbol, "Generator"), define(Gp, iteratorSymbol, function () { return this; }), define(Gp, "toString", function () { return "[object Generator]"; }), exports.keys = function (val) { var object = Object(val), keys = []; for (var key in object) { keys.push(key); } return keys.reverse(), function next() { for (; keys.length;) { var key = keys.pop(); if (key in object) return next.value = key, next.done = !1, next; } return next.done = !0, next; }; }, exports.values = values, Context.prototype = { constructor: Context, reset: function reset(skipTempReset) { if (this.prev = 0, this.next = 0, this.sent = this._sent = undefined, this.done = !1, this.delegate = null, this.method = "next", this.arg = undefined, this.tryEntries.forEach(resetTryEntry), !skipTempReset) for (var name in this) { "t" === name.charAt(0) && hasOwn.call(this, name) && !isNaN(+name.slice(1)) && (this[name] = undefined); } }, stop: function stop() { this.done = !0; var rootRecord = this.tryEntries[0].completion; if ("throw" === rootRecord.type) throw rootRecord.arg; return this.rval; }, dispatchException: function dispatchException(exception) { if (this.done) throw exception; var context = this; function handle(loc, caught) { return record.type = "throw", record.arg = exception, context.next = loc, caught && (context.method = "next", context.arg = undefined), !!caught; } for (var i = this.tryEntries.length - 1; i >= 0; --i) { var entry = this.tryEntries[i], record = entry.completion; if ("root" === entry.tryLoc) return handle("end"); if (entry.tryLoc <= this.prev) { var hasCatch = hasOwn.call(entry, "catchLoc"), hasFinally = hasOwn.call(entry, "finallyLoc"); if (hasCatch && hasFinally) { if (this.prev < entry.catchLoc) return handle(entry.catchLoc, !0); if (this.prev < entry.finallyLoc) return handle(entry.finallyLoc); } else if (hasCatch) { if (this.prev < entry.catchLoc) return handle(entry.catchLoc, !0); } else { if (!hasFinally) throw new Error("try statement without catch or finally"); if (this.prev < entry.finallyLoc) return handle(entry.finallyLoc); } } } }, abrupt: function abrupt(type, arg) { for (var i = this.tryEntries.length - 1; i >= 0; --i) { var entry = this.tryEntries[i]; if (entry.tryLoc <= this.prev && hasOwn.call(entry, "finallyLoc") && this.prev < entry.finallyLoc) { var finallyEntry = entry; break; } } finallyEntry && ("break" === type || "continue" === type) && finallyEntry.tryLoc <= arg && arg <= finallyEntry.finallyLoc && (finallyEntry = null); var record = finallyEntry ? finallyEntry.completion : {}; return record.type = type, record.arg = arg, finallyEntry ? (this.method = "next", this.next = finallyEntry.finallyLoc, ContinueSentinel) : this.complete(record); }, complete: function complete(record, afterLoc) { if ("throw" === record.type) throw record.arg; return "break" === record.type || "continue" === record.type ? this.next = record.arg : "return" === record.type ? (this.rval = this.arg = record.arg, this.method = "return", this.next = "end") : "normal" === record.type && afterLoc && (this.next = afterLoc), ContinueSentinel; }, finish: function finish(finallyLoc) { for (var i = this.tryEntries.length - 1; i >= 0; --i) { var entry = this.tryEntries[i]; if (entry.finallyLoc === finallyLoc) return this.complete(entry.completion, entry.afterLoc), resetTryEntry(entry), ContinueSentinel; } }, "catch": function _catch(tryLoc) { for (var i = this.tryEntries.length - 1; i >= 0; --i) { var entry = this.tryEntries[i]; if (entry.tryLoc === tryLoc) { var record = entry.completion; if ("throw" === record.type) { var thrown = record.arg; resetTryEntry(entry); } return thrown; } } throw new Error("illegal catch attempt"); }, delegateYield: function delegateYield(iterable, resultName, nextLoc) { return this.delegate = { iterator: values(iterable), resultName: resultName, nextLoc: nextLoc }, "next" === this.method && (this.arg = undefined), ContinueSentinel; } }, exports; }
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }
function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var Filter = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Filter, _Component);
  function Filter() {
    var _this;
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _Component.call.apply(_Component, [this].concat(args)) || this;
    _this.state = {
      openIndex: 0,
      modalVisible: false,
      selectedFilters: {},
      tabsBarHeight: 0,
      cascadeChildren: [],
      cascadeActive: 0,
      currentMultiSelected: []
    };
    _this.setVisible = function (visible) {
      if (visible === void 0) {
        visible = false;
      }
      var footer = document.querySelector('.template-footer');
      if (footer) {
        footer.style.zIndex = visible ? 0 : 1;
      }
      var masonryWrap = _this.props.masonryWrap;
      if (masonryWrap) {
        masonryWrap.style.minHeight = visible ? '100vh' : 'auto';
      }
      _this.setState({
        modalVisible: visible
      });
    };
    return _this;
  }
  var _proto = Filter.prototype;
  _proto.componentDidMount = function componentDidMount() {
    var _this$props = this.props,
      tabsEl = _this$props.tabsEl,
      navBarHeight = _this$props.navBarHeight;
    if (!tabsEl) return;
    this.setState({
      tabsBarHeight: tabsEl.clientHeight + navBarHeight
    });
  };
  _proto.initCascadeData = /*#__PURE__*/function () {
    var _initCascadeData = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee(index) {
      var currentList, _this$props2, filterData, filterSelected, currentOptions, selected, currentId, defaultChildren;
      return _regeneratorRuntime().wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              currentList = [];
              _this$props2 = this.props, filterData = _this$props2.filterData, filterSelected = _this$props2.filterSelected;
              currentOptions = filterData[index].options;
              selected = filterSelected[index];
              currentId = selected.parentId !== undefined ? selected.parentId : selected.value;
              defaultChildren = currentOptions.filter(function (v) {
                return v.value === currentId;
              })[0];
              if (!filterData[index].callBack) {
                _context.next = 12;
                break;
              }
              _context.next = 9;
              return filterData[index].callBack(defaultChildren);
            case 9:
              currentList = _context.sent;
              _context.next = 13;
              break;
            case 12:
              currentList = defaultChildren ? defaultChildren.children : [];
            case 13:
              this.setState({
                cascadeChildren: currentList
              });
            case 14:
            case "end":
              return _context.stop();
          }
        }
      }, _callee, this);
    }));
    function initCascadeData(_x) {
      return _initCascadeData.apply(this, arguments);
    }
    return initCascadeData;
  }();
  _proto.handClickPanel = function handClickPanel(item, index) {
    if (!item.options || item.options.length <= 0) return;
    var _this$state = this.state,
      modalVisible = _this$state.modalVisible,
      openIndex = _this$state.openIndex;
    item.type === 1 && this.initCascadeData(index);
    var visible = index === openIndex ? !modalVisible : true;
    this.setVisible(visible);
    this.setState({
      openIndex: index
    });
    var onClickPanel = this.props.onClickPanel;
    onClickPanel && onClickPanel(item, index);
  };
  _proto.handSelected = function handSelected(item) {
    var _extends2;
    var openIndex = this.state.openIndex;
    var _this$props3 = this.props,
      onChange = _this$props3.onChange,
      filterSelected = _this$props3.filterSelected;
    this.setVisible();
    var selectedFilters = _extends({}, filterSelected, (_extends2 = {}, _extends2[openIndex] = item, _extends2));
    onChange && onChange(selectedFilters);
  };
  _proto.cascadeClick = /*#__PURE__*/function () {
    var _cascadeClick = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee2(item, call) {
      var cascadeChildren;
      return _regeneratorRuntime().wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              cascadeChildren = [];
              if (!call.callBack) {
                _context2.next = 7;
                break;
              }
              _context2.next = 4;
              return call.callBack(item);
            case 4:
              cascadeChildren = _context2.sent;
              _context2.next = 8;
              break;
            case 7:
              cascadeChildren = item.children || [];
            case 8:
              this.setState({
                cascadeChildren: cascadeChildren,
                cascadeActive: item.value || 0
              });
            case 9:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2, this);
    }));
    function cascadeClick(_x2, _x3) {
      return _cascadeClick.apply(this, arguments);
    }
    return cascadeClick;
  }();
  _proto.cascadeSelected = function cascadeSelected(item) {
    var _extends3;
    var openIndex = this.state.openIndex;
    var _this$props4 = this.props,
      onChange = _this$props4.onChange,
      filterSelected = _this$props4.filterSelected;
    this.setVisible();
    var newItem = _extends({}, item, {
      name: item.label || item.name
    });
    var selectedFilters = _extends({}, filterSelected, (_extends3 = {}, _extends3[openIndex] = newItem, _extends3));
    onChange && onChange(selectedFilters);
  };
  _proto.multiSelect = function multiSelect(item) {
    var _this$state$currentMu = this.state.currentMultiSelected,
      currentMultiSelected = _this$state$currentMu === void 0 ? [] : _this$state$currentMu;
    var index = currentMultiSelected.findIndex(function (v) {
      return v.value === item.value;
    });
    if (index > -1) {
      currentMultiSelected.splice(index, 1);
    } else {
      currentMultiSelected.push(item);
    }
    this.setState({
      currentMultiSelected: currentMultiSelected.filter(function (v) {
        return v;
      })
    });
  };
  _proto.submitMultiSelect = function submitMultiSelect() {
    var _extends4;
    var _this$state$currentMu2 = this.state.currentMultiSelected,
      currentMultiSelected = _this$state$currentMu2 === void 0 ? [] : _this$state$currentMu2;
    var _this$props5 = this.props,
      _this$props5$filterSe = _this$props5.filterSelected,
      filterSelected = _this$props5$filterSe === void 0 ? {} : _this$props5$filterSe,
      onChange = _this$props5.onChange;
    var current = currentMultiSelected[0] || {}; // 目前只有一个代金券 下标直接写死
    var params = {
      value: Number(current.value) || 0,
      name: '筛选'
    };
    var openIndex = 3;
    var selectedFilters = _extends({}, filterSelected, (_extends4 = {}, _extends4[openIndex] = params, _extends4));
    this.setVisible();
    onChange && onChange(selectedFilters);
  };
  _proto.resetMultiSelect = function resetMultiSelect() {
    this.setState({
      currentMultiSelected: []
    });
  };
  _proto.render = function render() {
    var _this2 = this;
    var _this$props6 = this.props,
      _this$props6$filterDa = _this$props6.filterData,
      filterData = _this$props6$filterDa === void 0 ? [] : _this$props6$filterDa,
      _this$props6$filterSe = _this$props6.filterSelected,
      filterSelected = _this$props6$filterSe === void 0 ? {} : _this$props6$filterSe;
    var _this$state2 = this.state,
      openIndex = _this$state2.openIndex,
      modalVisible = _this$state2.modalVisible,
      tabsBarHeight = _this$state2.tabsBarHeight,
      cascadeChildren = _this$state2.cascadeChildren,
      cascadeActive = _this$state2.cascadeActive,
      _this$state2$currentM = _this$state2.currentMultiSelected,
      currentMultiSelected = _this$state2$currentM === void 0 ? [] : _this$state2$currentM;
    var currentSelected = filterSelected[openIndex] || {};
    var currentPanelData = filterData[openIndex];
    var options = currentPanelData.options || [];
    if (Object.prototype.toString.call(filterData) !== '[object array]' && filterData.length <= 0) {
      throw new Error('filters组件的filterData属性只能接收object类型');
    }
    var MultipleModal = function MultipleModal() {
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "filter-content"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: "coupon-title"
      }, "\u4F18\u60E0 (\u53EF\u591A\u9009)"), /*#__PURE__*/_react["default"].createElement("div", {
        className: "coupon-select"
      }, options.map(function (item) {
        return /*#__PURE__*/_react["default"].createElement("div", {
          key: item.value,
          onClick: function onClick() {
            return _this2.multiSelect(item);
          },
          className: currentMultiSelected.some(function (v) {
            return v.value === item.value;
          }) ? 'coupon-select-item selected' : 'coupon-select-item'
        }, currentMultiSelected.some(function (v) {
          return v.value === item.value;
        }) && /*#__PURE__*/_react["default"].createElement("img", {
          className: "img",
          src: "https://bosscdn.otosaas.com/test/20220815/20220815182656selected_icon_01.png?path=0,485,488",
          alt: ""
        }), item.name);
      })), /*#__PURE__*/_react["default"].createElement("div", {
        className: "coupon-footer"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: "coupon-footer-item reset",
        onClick: function onClick() {
          return _this2.resetMultiSelect();
        }
      }, "\u91CD\u7F6E"), /*#__PURE__*/_react["default"].createElement("div", {
        className: "coupon-footer-item affirm",
        onClick: function onClick() {
          return _this2.submitMultiSelect();
        }
      }, "\u786E\u8BA4")));
    };
    var CascadeModal = function CascadeModal() {
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "filter-content"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: "cascade"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: "cascade-lf cascade-item"
      }, options.map(function (item) {
        return /*#__PURE__*/_react["default"].createElement("div", {
          key: item.value,
          className: cascadeActive === item.value ? 'item selected' : 'item',
          onClick: function onClick() {
            return _this2.cascadeClick(item, currentPanelData);
          }
        }, item.name);
      })), /*#__PURE__*/_react["default"].createElement("div", {
        className: "cascade-rt cascade-item"
      }, cascadeChildren.map(function (item) {
        return /*#__PURE__*/_react["default"].createElement("div", {
          key: item.value,
          className: currentSelected.value === item.value ? 'item selected' : 'item',
          onClick: function onClick() {
            return _this2.cascadeSelected(item);
          }
        }, item.name, currentSelected.value === item.value && /*#__PURE__*/_react["default"].createElement("img", {
          src: _selected["default"],
          alt: ""
        }));
      }))));
    };
    var DefaultModal = function DefaultModal() {
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "filter-content filter-default-modal"
      }, options.map(function (item) {
        return /*#__PURE__*/_react["default"].createElement("div", {
          key: item.value,
          className: currentSelected.value === item.value ? 'item selected' : 'item',
          onClick: function onClick() {
            return _this2.handSelected(item);
          }
        }, item.name, " ", currentSelected.value === item.value && /*#__PURE__*/_react["default"].createElement("img", {
          src: _selected["default"],
          alt: ""
        }));
      }));
    };
    var FilterModal = function FilterModal() {
      return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, currentPanelData.type === 1 ? /*#__PURE__*/_react["default"].createElement(CascadeModal, null) : currentPanelData.type == 2 ? /*#__PURE__*/_react["default"].createElement(MultipleModal, null) : /*#__PURE__*/_react["default"].createElement(DefaultModal, null));
    };
    return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("div", {
      className: "filter-panel"
    }, filterData.map(function (item, index) {
      return /*#__PURE__*/_react["default"].createElement("div", {
        key: item.name,
        className: filterSelected[index] && filterSelected[index].value ? 'item open' : 'item',
        onClick: function onClick() {
          return _this2.handClickPanel(item, index);
        }
      }, filterSelected[index] && filterSelected[index].name || item.name, /*#__PURE__*/_react["default"].createElement("img", {
        className: "icon-iconxiala",
        src: "https://bosscdn.otosaas.com/test/20220815/202208151448524821660545167_.pic.jpg?path=0,485,486"
      }));
    })), /*#__PURE__*/_react["default"].createElement(_antdMobileV.Mask, {
      visible: modalVisible,
      onMaskClick: function onMaskClick() {
        return _this2.setVisible(false);
      },
      style: {
        top: tabsBarHeight
      }
    }, /*#__PURE__*/_react["default"].createElement(FilterModal, null)));
  };
  return Filter;
}(_react.Component);
Filter.defaultProps = {
  navBarHeight: 0
};
var _default = Filter;
exports["default"] = _default;
