// @kuping/oto_component/lib/business/MasonryLayouts/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var _antdMobileV = require("antd-mobile-v5");
var _const = require("../../utils/const");
var _common = require("../../utils/common");
var _dictMap = _interopRequireDefault(require("../../utils/dictMap.json"));
var _asyncFun = require("./asyncFun");
var _SkeletonMasonry = _interopRequireDefault(require("./SkeletonMasonry"));
var _Commodity = _interopRequireDefault(require("../Items/Commodity"));
var _Coupon = _interopRequireDefault(require("../Items/Coupon"));
var _Shop = _interopRequireDefault(require("../Items/Shop"));
var _Tickets = _interopRequireDefault(require("../Items/Tickets"));
var _Masonry = _interopRequireDefault(require("../Masonry"));
var _Filter = _interopRequireDefault(require("./Filter"));
var _Theme = _interopRequireDefault(require("./Theme"));
var _NoneConfig = _interopRequireDefault(require("../NoneConfig"));
require("./index.css");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _regeneratorRuntime() { "use strict"; /*! regenerator-runtime -- Copyright (c) 2014-present, Facebook, Inc. -- license (MIT): https://github.com/facebook/regenerator/blob/main/LICENSE */ _regeneratorRuntime = function _regeneratorRuntime() { return exports; }; var exports = {}, Op = Object.prototype, hasOwn = Op.hasOwnProperty, defineProperty = Object.defineProperty || function (obj, key, desc) { obj[key] = desc.value; }, $Symbol = "function" == typeof Symbol ? Symbol : {}, iteratorSymbol = $Symbol.iterator || "@@iterator", asyncIteratorSymbol = $Symbol.asyncIterator || "@@asyncIterator", toStringTagSymbol = $Symbol.toStringTag || "@@toStringTag"; function define(obj, key, value) { return Object.defineProperty(obj, key, { value: value, enumerable: !0, configurable: !0, writable: !0 }), obj[key]; } try { define({}, ""); } catch (err) { define = function define(obj, key, value) { return obj[key] = value; }; } function wrap(innerFn, outerFn, self, tryLocsList) { var protoGenerator = outerFn && outerFn.prototype instanceof Generator ? outerFn : Generator, generator = Object.create(protoGenerator.prototype), context = new Context(tryLocsList || []); return defineProperty(generator, "_invoke", { value: makeInvokeMethod(innerFn, self, context) }), generator; } function tryCatch(fn, obj, arg) { try { return { type: "normal", arg: fn.call(obj, arg) }; } catch (err) { return { type: "throw", arg: err }; } } exports.wrap = wrap; var ContinueSentinel = {}; function Generator() {} function GeneratorFunction() {} function GeneratorFunctionPrototype() {} var IteratorPrototype = {}; define(IteratorPrototype, iteratorSymbol, function () { return this; }); var getProto = Object.getPrototypeOf, NativeIteratorPrototype = getProto && getProto(getProto(values([]))); NativeIteratorPrototype && NativeIteratorPrototype !== Op && hasOwn.call(NativeIteratorPrototype, iteratorSymbol) && (IteratorPrototype = NativeIteratorPrototype); var Gp = GeneratorFunctionPrototype.prototype = Generator.prototype = Object.create(IteratorPrototype); function defineIteratorMethods(prototype) { ["next", "throw", "return"].forEach(function (method) { define(prototype, method, function (arg) { return this._invoke(method, arg); }); }); } function AsyncIterator(generator, PromiseImpl) { function invoke(method, arg, resolve, reject) { var record = tryCatch(generator[method], generator, arg); if ("throw" !== record.type) { var result = record.arg, value = result.value; return value && "object" == _typeof(value) && hasOwn.call(value, "__await") ? PromiseImpl.resolve(value.__await).then(function (value) { invoke("next", value, resolve, reject); }, function (err) { invoke("throw", err, resolve, reject); }) : PromiseImpl.resolve(value).then(function (unwrapped) { result.value = unwrapped, resolve(result); }, function (error) { return invoke("throw", error, resolve, reject); }); } reject(record.arg); } var previousPromise; defineProperty(this, "_invoke", { value: function value(method, arg) { function callInvokeWithMethodAndArg() { return new PromiseImpl(function (resolve, reject) { invoke(method, arg, resolve, reject); }); } return previousPromise = previousPromise ? previousPromise.then(callInvokeWithMethodAndArg, callInvokeWithMethodAndArg) : callInvokeWithMethodAndArg(); } }); } function makeInvokeMethod(innerFn, self, context) { var state = "suspendedStart"; return function (method, arg) { if ("executing" === state) throw new Error("Generator is already running"); if ("completed" === state) { if ("throw" === method) throw arg; return doneResult(); } for (context.method = method, context.arg = arg;;) { var delegate = context.delegate; if (delegate) { var delegateResult = maybeInvokeDelegate(delegate, context); if (delegateResult) { if (delegateResult === ContinueSentinel) continue; return delegateResult; } } if ("next" === context.method) context.sent = context._sent = context.arg;else if ("throw" === context.method) { if ("suspendedStart" === state) throw state = "completed", context.arg; context.dispatchException(context.arg); } else "return" === context.method && context.abrupt("return", context.arg); state = "executing"; var record = tryCatch(innerFn, self, context); if ("normal" === record.type) { if (state = context.done ? "completed" : "suspendedYield", record.arg === ContinueSentinel) continue; return { value: record.arg, done: context.done }; } "throw" === record.type && (state = "completed", context.method = "throw", context.arg = record.arg); } }; } function maybeInvokeDelegate(delegate, context) { var method = delegate.iterator[context.method]; if (undefined === method) { if (context.delegate = null, "throw" === context.method) { if (delegate.iterator["return"] && (context.method = "return", context.arg = undefined, maybeInvokeDelegate(delegate, context), "throw" === context.method)) return ContinueSentinel; context.method = "throw", context.arg = new TypeError("The iterator does not provide a 'throw' method"); } return ContinueSentinel; } var record = tryCatch(method, delegate.iterator, context.arg); if ("throw" === record.type) return context.method = "throw", context.arg = record.arg, context.delegate = null, ContinueSentinel; var info = record.arg; return info ? info.done ? (context[delegate.resultName] = info.value, context.next = delegate.nextLoc, "return" !== context.method && (context.method = "next", context.arg = undefined), context.delegate = null, ContinueSentinel) : info : (context.method = "throw", context.arg = new TypeError("iterator result is not an object"), context.delegate = null, ContinueSentinel); } function pushTryEntry(locs) { var entry = { tryLoc: locs[0] }; 1 in locs && (entry.catchLoc = locs[1]), 2 in locs && (entry.finallyLoc = locs[2], entry.afterLoc = locs[3]), this.tryEntries.push(entry); } function resetTryEntry(entry) { var record = entry.completion || {}; record.type = "normal", delete record.arg, entry.completion = record; } function Context(tryLocsList) { this.tryEntries = [{ tryLoc: "root" }], tryLocsList.forEach(pushTryEntry, this), this.reset(!0); } function values(iterable) { if (iterable) { var iteratorMethod = iterable[iteratorSymbol]; if (iteratorMethod) return iteratorMethod.call(iterable); if ("function" == typeof iterable.next) return iterable; if (!isNaN(iterable.length)) { var i = -1, next = function next() { for (; ++i < iterable.length;) { if (hasOwn.call(iterable, i)) return next.value = iterable[i], next.done = !1, next; } return next.value = undefined, next.done = !0, next; }; return next.next = next; } } return { next: doneResult }; } function doneResult() { return { value: undefined, done: !0 }; } return GeneratorFunction.prototype = GeneratorFunctionPrototype, defineProperty(Gp, "constructor", { value: GeneratorFunctionPrototype, configurable: !0 }), defineProperty(GeneratorFunctionPrototype, "constructor", { value: GeneratorFunction, configurable: !0 }), GeneratorFunction.displayName = define(GeneratorFunctionPrototype, toStringTagSymbol, "GeneratorFunction"), exports.isGeneratorFunction = function (genFun) { var ctor = "function" == typeof genFun && genFun.constructor; return !!ctor && (ctor === GeneratorFunction || "GeneratorFunction" === (ctor.displayName || ctor.name)); }, exports.mark = function (genFun) { return Object.setPrototypeOf ? Object.setPrototypeOf(genFun, GeneratorFunctionPrototype) : (genFun.__proto__ = GeneratorFunctionPrototype, define(genFun, toStringTagSymbol, "GeneratorFunction")), genFun.prototype = Object.create(Gp), genFun; }, exports.awrap = function (arg) { return { __await: arg }; }, defineIteratorMethods(AsyncIterator.prototype), define(AsyncIterator.prototype, asyncIteratorSymbol, function () { return this; }), exports.AsyncIterator = AsyncIterator, exports.async = function (innerFn, outerFn, self, tryLocsList, PromiseImpl) { void 0 === PromiseImpl && (PromiseImpl = Promise); var iter = new AsyncIterator(wrap(innerFn, outerFn, self, tryLocsList), PromiseImpl); return exports.isGeneratorFunction(outerFn) ? iter : iter.next().then(function (result) { return result.done ? result.value : iter.next(); }); }, defineIteratorMethods(Gp), define(Gp, toStringTagSymbol, "Generator"), define(Gp, iteratorSymbol, function () { return this; }), define(Gp, "toString", function () { return "[object Generator]"; }), exports.keys = function (val) { var object = Object(val), keys = []; for (var key in object) { keys.push(key); } return keys.reverse(), function next() { for (; keys.length;) { var key = keys.pop(); if (key in object) return next.value = key, next.done = !1, next; } return next.done = !0, next; }; }, exports.values = values, Context.prototype = { constructor: Context, reset: function reset(skipTempReset) { if (this.prev = 0, this.next = 0, this.sent = this._sent = undefined, this.done = !1, this.delegate = null, this.method = "next", this.arg = undefined, this.tryEntries.forEach(resetTryEntry), !skipTempReset) for (var name in this) { "t" === name.charAt(0) && hasOwn.call(this, name) && !isNaN(+name.slice(1)) && (this[name] = undefined); } }, stop: function stop() { this.done = !0; var rootRecord = this.tryEntries[0].completion; if ("throw" === rootRecord.type) throw rootRecord.arg; return this.rval; }, dispatchException: function dispatchException(exception) { if (this.done) throw exception; var context = this; function handle(loc, caught) { return record.type = "throw", record.arg = exception, context.next = loc, caught && (context.method = "next", context.arg = undefined), !!caught; } for (var i = this.tryEntries.length - 1; i >= 0; --i) { var entry = this.tryEntries[i], record = entry.completion; if ("root" === entry.tryLoc) return handle("end"); if (entry.tryLoc <= this.prev) { var hasCatch = hasOwn.call(entry, "catchLoc"), hasFinally = hasOwn.call(entry, "finallyLoc"); if (hasCatch && hasFinally) { if (this.prev < entry.catchLoc) return handle(entry.catchLoc, !0); if (this.prev < entry.finallyLoc) return handle(entry.finallyLoc); } else if (hasCatch) { if (this.prev < entry.catchLoc) return handle(entry.catchLoc, !0); } else { if (!hasFinally) throw new Error("try statement without catch or finally"); if (this.prev < entry.finallyLoc) return handle(entry.finallyLoc); } } } }, abrupt: function abrupt(type, arg) { for (var i = this.tryEntries.length - 1; i >= 0; --i) { var entry = this.tryEntries[i]; if (entry.tryLoc <= this.prev && hasOwn.call(entry, "finallyLoc") && this.prev < entry.finallyLoc) { var finallyEntry = entry; break; } } finallyEntry && ("break" === type || "continue" === type) && finallyEntry.tryLoc <= arg && arg <= finallyEntry.finallyLoc && (finallyEntry = null); var record = finallyEntry ? finallyEntry.completion : {}; return record.type = type, record.arg = arg, finallyEntry ? (this.method = "next", this.next = finallyEntry.finallyLoc, ContinueSentinel) : this.complete(record); }, complete: function complete(record, afterLoc) { if ("throw" === record.type) throw record.arg; return "break" === record.type || "continue" === record.type ? this.next = record.arg : "return" === record.type ? (this.rval = this.arg = record.arg, this.method = "return", this.next = "end") : "normal" === record.type && afterLoc && (this.next = afterLoc), ContinueSentinel; }, finish: function finish(finallyLoc) { for (var i = this.tryEntries.length - 1; i >= 0; --i) { var entry = this.tryEntries[i]; if (entry.finallyLoc === finallyLoc) return this.complete(entry.completion, entry.afterLoc), resetTryEntry(entry), ContinueSentinel; } }, "catch": function _catch(tryLoc) { for (var i = this.tryEntries.length - 1; i >= 0; --i) { var entry = this.tryEntries[i]; if (entry.tryLoc === tryLoc) { var record = entry.completion; if ("throw" === record.type) { var thrown = record.arg; resetTryEntry(entry); } return thrown; } } throw new Error("illegal catch attempt"); }, delegateYield: function delegateYield(iterable, resultName, nextLoc) { return this.delegate = { iterator: values(iterable), resultName: resultName, nextLoc: nextLoc }, "next" === this.method && (this.arg = undefined), ContinueSentinel; } }, exports; }
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }
function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
var MasonryLayout = /*#__PURE__*/function (_Component) {
  _inheritsLoose(MasonryLayout, _Component);
  function MasonryLayout() {
    var _this;
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }
    _this = _Component.call.apply(_Component, [this].concat(args)) || this;
    _this.state = {
      activeIndex: 0,
      listData: [],
      hasMore: true,
      loading: true,
      recordRequest: {
        pageNum: 0
      },
      jingdianSelected: _const.defaultSelectedFilter,
      filterShopSelected: _const.defaultShopSelectedFilter,
      themeData: [],
      categoryList: [],
      currentDict: [],
      navBarHeight: 0
    };
    _this.ThemeComponent = null;
    _this.swiperRef = null;
    _this.masonryWrap = null;
    _this.tabsEl = null;
    _this.filterVisible = 1;
    _this.type = 0;
    _this.code = '';
    _this.sort = 1;
    return _this;
  }
  var _proto = MasonryLayout.prototype;
  _proto.componentDidMount = /*#__PURE__*/function () {
    var _componentDidMount = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee() {
      var _data$recommendCode,
        _data$recommendCode2,
        _this2 = this;
      var _this$props, appCode, locationInfo, scrollContainer, dataSource, activeIndex, data, bodyEl, isShop, isJingdian;
      return _regeneratorRuntime().wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              this.getNavBarHeight();
              _this$props = this.props, appCode = _this$props.appCode, locationInfo = _this$props.locationInfo, scrollContainer = _this$props.scrollContainer;
              dataSource = this.props.data.dataSource;
              activeIndex = this.state.activeIndex;
              data = dataSource ? dataSource[activeIndex] : {
                recommendType: 0,
                recommendCode: {
                  recommendCode: ''
                },
                filter: 0,
                sort: 1
              };
              this.type = data.recommendType;
              this.sort = data.sort;
              this.code = ((_data$recommendCode = data.recommendCode) === null || _data$recommendCode === void 0 ? void 0 : _data$recommendCode.recommendCode) || ((_data$recommendCode2 = data.recommendCode) === null || _data$recommendCode2 === void 0 ? void 0 : _data$recommendCode2.id) || '';
              this.filterVisible = data.filter;
              if (scrollContainer) {
                scrollContainer.addEventListener('scroll', function () {
                  return _this2.scroll();
                });
              } else {
                bodyEl = document.querySelector('#root');
                bodyEl && bodyEl.addEventListener('scroll', function () {
                  return _this2.scroll();
                });
              }
              this.initState(0);
              isShop = dataSource.find(function (v) {
                return v.recommendType === 7;
              });
              isJingdian = dataSource.find(function (v) {
                return v.recommendType === 6;
              });
              isJingdian && this.getThemeData();
              _context.t0 = this;
              if (!isShop) {
                _context.next = 21;
                break;
              }
              _context.next = 18;
              return (0, _asyncFun.getAllCategory)(appCode);
            case 18:
              _context.t1 = _context.sent;
              _context.next = 22;
              break;
            case 21:
              _context.t1 = [];
            case 22:
              _context.t2 = _context.t1;
              _context.t3 = (0, _asyncFun.filterCurrentCity)({
                cityArr: _dictMap["default"].data,
                locationInfo: locationInfo
              });
              _context.t4 = {
                categoryList: _context.t2,
                currentDict: _context.t3
              };
              _context.t0.setState.call(_context.t0, _context.t4);
            case 26:
            case "end":
              return _context.stop();
          }
        }
      }, _callee, this);
    }));
    function componentDidMount() {
      return _componentDidMount.apply(this, arguments);
    }
    return componentDidMount;
  }();
  _proto.getNavBarHeight = function getNavBarHeight() {
    var navBarHeight = 0;
    if ((0, _common.getIsTopNavBar)()) {
      var title = document.getElementById('title');
      navBarHeight = title ? title.clientHeight : 44;
    }
    this.setState({
      navBarHeight: navBarHeight
    });
  };
  _proto.scroll = function scroll() {
    var navBarHeight = this.state.navBarHeight;
    if (!this.tabsEl) return;
    var bounding = this.tabsEl.getBoundingClientRect();
    if (bounding.top <= navBarHeight + 2) {
      var stickyStyle = this.props.data.basicAttribute.stickyStyle;
      this.tabsEl.style.backgroundColor = stickyStyle.backgroundColor;
    } else {
      this.tabsEl.style.backgroundColor = '';
    }
  };
  _proto.getThemeData = /*#__PURE__*/function () {
    var _getThemeData = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee2() {
      var _this$props2, locationInfo, appCode, _ref, addressComponent, data;
      return _regeneratorRuntime().wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              _this$props2 = this.props, locationInfo = _this$props2.locationInfo, appCode = _this$props2.appCode;
              _ref = locationInfo || {}, addressComponent = _ref.addressComponent;
              _context2.next = 4;
              return (0, _asyncFun.getMenpiaoThemes)(addressComponent.city, appCode);
            case 4:
              data = _context2.sent;
              this.setState({
                themeData: data
              });
              _const.filterMenpiaoData[0].options = data;
            case 7:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2, this);
    }));
    function getThemeData() {
      return _getThemeData.apply(this, arguments);
    }
    return getThemeData;
  }();
  _proto.loadingDelayClone = function loadingDelayClone() {
    var _this3 = this;
    setTimeout(function () {
      _this3.setState({
        loading: false
      });
    }, 500);
  };
  _proto.initState = function initState(index, isTop) {
    var _current$recommendCod,
      _current$recommendCod2,
      _this4 = this;
    if (isTop === void 0) {
      isTop = false;
    }
    var dataSource = this.props.data.dataSource;
    var current = dataSource[index];
    var code = ((_current$recommendCod = current.recommendCode) === null || _current$recommendCod === void 0 ? void 0 : _current$recommendCod.recommendCode) || ((_current$recommendCod2 = current.recommendCode) === null || _current$recommendCod2 === void 0 ? void 0 : _current$recommendCod2.id) || '';
    var type = current.recommendType;
    this.filterVisible = current.filter;
    this.code = code;
    this.sort = current.sort;
    this.type = type;
    this.setState({
      activeIndex: index,
      listData: [],
      hasMore: true,
      loading: true,
      jingdianSelected: _const.defaultSelectedFilter,
      filterShopSelected: _const.defaultShopSelectedFilter,
      recordRequest: {
        pageNum: 0
      }
    }, function () {
      _this4.loadMore();
      isTop && _this4.setTabsTop();
    });
  };
  _proto.setTabsTop = function setTabsTop() {
    if (!this.tabsEl) return;
    var scrollContainer = this.props.scrollContainer;
    var offsetTop = this.tabsEl.offsetTop;
    var container = scrollContainer;
    if (!scrollContainer) {
      container = document.querySelector('#root');
    }
    container.scrollTo(0, offsetTop);
  };
  _proto.tabsChange = function tabsChange(key) {
    var _this$swiperRef;
    var dataSource = this.props.data.dataSource;
    var index = dataSource.findIndex(function (_, index) {
      return index + '' === key;
    });
    (_this$swiperRef = this.swiperRef) === null || _this$swiperRef === void 0 ? void 0 : _this$swiperRef.swipeTo(index);
  };
  _proto.onIndexChange = function onIndexChange(index) {
    this.initState(index, true);
  };
  _proto.loadMore = /*#__PURE__*/function () {
    var _loadMore = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee3() {
      var val, _this$props3, locationInfo, appCode, options, res, newData;
      return _regeneratorRuntime().wrap(function _callee3$(_context3) {
        while (1) {
          switch (_context3.prev = _context3.next) {
            case 0:
              val = this.state.listData;
              _this$props3 = this.props, locationInfo = _this$props3.locationInfo, appCode = _this$props3.appCode;
              options = {
                code: this.code,
                type: this.type,
                sort: this.sort,
                appCode: appCode,
                pageNum: this.state.recordRequest.pageNum,
                locationInfo: locationInfo
              };
              _context3.prev = 3;
              _context3.next = 6;
              return (0, _asyncFun._initListData)(options);
            case 6:
              res = _context3.sent;
              if (res) {
                newData = [].concat(val, res.dataSource);
                this.setState({
                  listData: newData,
                  recordRequest: res,
                  hasMore: newData.length < res.total
                });
              } else {
                this.setState({
                  hasMore: true
                });
              }
              this.loadingDelayClone();
              _context3.next = 14;
              break;
            case 11:
              _context3.prev = 11;
              _context3.t0 = _context3["catch"](3);
              this.loadingDelayClone();
            case 14:
            case "end":
              return _context3.stop();
          }
        }
      }, _callee3, this, [[3, 11]]);
    }));
    function loadMore() {
      return _loadMore.apply(this, arguments);
    }
    return loadMore;
  }();
  _proto.filterShopChange = /*#__PURE__*/function () {
    var _filterShopChange = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee4(filter) {
      var _this$props4, locationInfo, appCode, res;
      return _regeneratorRuntime().wrap(function _callee4$(_context4) {
        while (1) {
          switch (_context4.prev = _context4.next) {
            case 0:
              _this$props4 = this.props, locationInfo = _this$props4.locationInfo, appCode = _this$props4.appCode;
              _context4.prev = 1;
              _context4.next = 4;
              return (0, _asyncFun.getShop)({
                pageNum: 1,
                appCode: appCode,
                locationInfo: locationInfo,
                filterShopSelected: filter
              });
            case 4:
              res = _context4.sent;
              this.loadingDelayClone();
              if (res) {
                _context4.next = 8;
                break;
              }
              return _context4.abrupt("return");
            case 8:
              this.setState({
                listData: res.dataSource,
                recordRequest: res,
                hasMore: res.dataSource < res.total,
                filterShopSelected: filter
              });
              _context4.next = 14;
              break;
            case 11:
              _context4.prev = 11;
              _context4.t0 = _context4["catch"](1);
              this.loadingDelayClone();
            case 14:
            case "end":
              return _context4.stop();
          }
        }
      }, _callee4, this, [[1, 11]]);
    }));
    function filterShopChange(_x) {
      return _filterShopChange.apply(this, arguments);
    }
    return filterShopChange;
  }();
  _proto.filterChange = /*#__PURE__*/function () {
    var _filterChange = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee5(filter) {
      var _this5 = this;
      var _this$props5, locationInfo, appCode, res;
      return _regeneratorRuntime().wrap(function _callee5$(_context5) {
        while (1) {
          switch (_context5.prev = _context5.next) {
            case 0:
              if (filter === void 0) {
                filter = {};
              }
              _this$props5 = this.props, locationInfo = _this$props5.locationInfo, appCode = _this$props5.appCode;
              _context5.prev = 2;
              _context5.next = 5;
              return (0, _asyncFun.getMenpiaoList)({
                appCode: appCode,
                pageNum: 1,
                locationInfo: locationInfo,
                jingdianSelected: filter
              });
            case 5:
              res = _context5.sent;
              this.loadingDelayClone();
              if (res) {
                _context5.next = 9;
                break;
              }
              return _context5.abrupt("return");
            case 9:
              this.setState({
                listData: res.dataSource,
                recordRequest: res,
                hasMore: res.dataSource < res.total,
                jingdianSelected: filter
              }, function () {
                if (_this5.ThemeComponent) {
                  var selectedEl = _this5.ThemeComponent.selectedEl;
                  var themeWrapEl = _this5.ThemeComponent.themeWrapEl;
                  _this5.autoScrollCenter(selectedEl, themeWrapEl);
                }
              });
              _context5.next = 15;
              break;
            case 12:
              _context5.prev = 12;
              _context5.t0 = _context5["catch"](2);
              this.loadingDelayClone();
            case 15:
            case "end":
              return _context5.stop();
          }
        }
      }, _callee5, this, [[2, 12]]);
    }));
    function filterChange(_x2) {
      return _filterChange.apply(this, arguments);
    }
    return filterChange;
  }();
  _proto.autoScrollCenter = function autoScrollCenter(selectedEl, themeWrapEl) {
    if (!selectedEl || !themeWrapEl) return;
    var left = selectedEl.offsetLeft;
    var width = selectedEl.offsetWidth / 2;
    var innerWidth = window.innerWidth / 2;
    themeWrapEl.scrollLeft = left - innerWidth + width;
  };
  _proto.themeSelected = function themeSelected(v) {
    var _this6 = this;
    var jingdianSelected = this.state.jingdianSelected;
    var filter = _extends({}, jingdianSelected, {
      '0': v
    });
    this.setState({
      jingdianSelected: filter
    }, function () {
      _this6.filterChange(filter);
    });
  };
  _proto.render = function render() {
    var _this7 = this;
    var _this$props6 = this.props,
      appCode = _this$props6.appCode,
      emptyDataVisible = _this$props6.emptyDataVisible;
    var _this$props$data = this.props.data,
      dataSource = _this$props$data.dataSource,
      _this$props$data$basi = _this$props$data.basicAttribute,
      color = _this$props$data$basi.color,
      activeLineStyle = _this$props$data$basi.activeLineStyle,
      activeStyle = _this$props$data$basi.activeStyle,
      relationStyle = _this$props$data$basi.relationStyle,
      monitor = _this$props$data$basi.monitor,
      saleTipText = _this$props$data$basi.saleTipText,
      priceType = _this$props$data$basi.priceType;
    var _this$state = this.state,
      activeIndex = _this$state.activeIndex,
      listData = _this$state.listData,
      hasMore = _this$state.hasMore,
      loading = _this$state.loading,
      jingdianSelected = _this$state.jingdianSelected,
      filterShopSelected = _this$state.filterShopSelected,
      themeData = _this$state.themeData,
      currentDict = _this$state.currentDict,
      navBarHeight = _this$state.navBarHeight;
    if (monitor) monitor.componentName = '瀑布流';
    var selected = jingdianSelected['0'] || {
      name: '全部景点'
    };
    var filterShopData = [{
      name: '全城',
      type: 1,
      options: currentDict,
      callBack: function () {
        var _callBack = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee6(obj) {
          var data;
          return _regeneratorRuntime().wrap(function _callee6$(_context6) {
            while (1) {
              switch (_context6.prev = _context6.next) {
                case 0:
                  _context6.next = 2;
                  return (0, _asyncFun.areaData)(obj, appCode);
                case 2:
                  data = _context6.sent;
                  return _context6.abrupt("return", data);
                case 4:
                case "end":
                  return _context6.stop();
              }
            }
          }, _callee6);
        }));
        function callBack(_x3) {
          return _callBack.apply(this, arguments);
        }
        return callBack;
      }()
    }, {
      name: '智能排序',
      options: [{
        name: '智能排序',
        value: 3
      }, {
        name: '好评优先',
        value: 4
      }, {
        name: '距离优先',
        value: 1
      }, {
        name: '销量优先',
        value: 2
      }]
    }, {
      name: '全部分类',
      value: 0,
      type: 1,
      options: this.state.categoryList
    }, {
      name: '筛选',
      type: 2,
      options: [
      // {
      //   name: '买单优惠',
      //   value: 1,
      // },
      {
        name: '代金券',
        value: 2
      }
      // {
      //   name: '套餐券',
      //   value: 3,
      // },
      ]
    }];

    var RenderContent = function RenderContent(_ref2) {
      var data = _ref2.data,
        recommendType = _ref2.recommendType;
      if (recommendType === 4) return /*#__PURE__*/_react["default"].createElement(_Commodity["default"], {
        data: data,
        saleTipText: saleTipText,
        monitor: monitor,
        priceType: priceType
      });
      if (recommendType === 5) return /*#__PURE__*/_react["default"].createElement(_Coupon["default"], {
        data: data,
        monitor: monitor
      });
      if (recommendType === 6) return /*#__PURE__*/_react["default"].createElement(_Tickets["default"], {
        data: data,
        monitor: monitor
      });
      if (recommendType === 7) return /*#__PURE__*/_react["default"].createElement(_Shop["default"], {
        data: data,
        monitor: monitor,
        appCode: appCode
      });
      return /*#__PURE__*/_react["default"].createElement("span", {
        "data-msg": '未匹配到'
      });
    };
    var DefaultImg = function DefaultImg() {
      return /*#__PURE__*/_react["default"].createElement("img", {
        src: "https://bosscdn.otosaas.com/test/20221108/20221108160306%E7%80%91%E5%B8%83%E6%B5%81.png?path=0,650,664",
        style: {
          width: '100%',
          height: 'auto'
        }
      });
    };
    if (!dataSource.length) return emptyDataVisible ? /*#__PURE__*/_react["default"].createElement(DefaultImg, null) : /*#__PURE__*/_react["default"].createElement("span", {
      "data-msg": "dataSource\u4E3A\u7A7A"
    });
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: "template-masonry-layouts"
    }, /*#__PURE__*/_react["default"].createElement("div", {
      ref: function ref(el) {
        return _this7.tabsEl = el;
      },
      className: relationStyle === 2 ? 'tabs-double-title' : '',
      style: {
        position: 'sticky',
        top: '0',
        zIndex: 1
      }
    }, /*#__PURE__*/_react["default"].createElement(_antdMobileV.Tabs, {
      activeKey: activeIndex + '',
      onChange: function onChange(key) {
        return _this7.tabsChange(key);
      },
      activeLineMode: "fixed",
      style: {
        color: color,
        '--fixed-active-line-width': '0.4rem',
        '--active-line-color': activeLineStyle.backgroundColor,
        '--active-title-color': activeStyle.color,
        '--active-line-height': '0.06rem'
      }
    }, dataSource.map(function (_ref3, index) {
      var name = _ref3.name,
        subtitle = _ref3.subtitle;
      return relationStyle !== 0 && /*#__PURE__*/_react["default"].createElement(_antdMobileV.Tabs.Tab, {
        key: index + '',
        title: /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("div", {
          className: "tab-title"
        }, name), relationStyle === 2 && /*#__PURE__*/_react["default"].createElement("div", {
          className: "tab-sub-title"
        }, subtitle))
      });
    })), this.type === 6 && this.filterVisible !== 0 && /*#__PURE__*/_react["default"].createElement(_Filter["default"], {
      filterData: _const.filterMenpiaoData,
      filterSelected: jingdianSelected,
      onChange: function onChange(filter) {
        return _this7.filterChange(filter);
      },
      onClickPanel: function onClickPanel() {
        return _this7.setTabsTop();
      },
      tabsEl: this.tabsEl,
      navBarHeight: navBarHeight,
      masonryWrap: this.masonryWrap
    }), this.type === 7 && this.filterVisible !== 0 && /*#__PURE__*/_react["default"].createElement(_Filter["default"], {
      filterData: filterShopData,
      filterSelected: filterShopSelected,
      onChange: function onChange(filter) {
        return _this7.filterShopChange(filter);
      },
      onClickPanel: function onClickPanel() {
        return _this7.setTabsTop();
      },
      tabsEl: this.tabsEl,
      navBarHeight: navBarHeight,
      masonryWrap: this.masonryWrap
    })), emptyDataVisible ?
    /*#__PURE__*/
    // relationStyle === 0 ? (
    //   <div style={{ padding: '20px', textAlign: 'center' }}>已配置无标题瀑布流</div>
    // ) : (
    //   <span />
    // )
    _react["default"].createElement(DefaultImg, null) : /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, this.type === 6 && /*#__PURE__*/_react["default"].createElement(_Theme["default"], {
      ref: function ref(el) {
        return _this7.ThemeComponent = el;
      },
      dataSource: themeData,
      selected: selected,
      onChange: function onChange(v) {
        return _this7.themeSelected(v);
      }
    }), /*#__PURE__*/_react["default"].createElement(_antdMobileV.Swiper, {
      direction: "horizontal",
      loop: true,
      indicator: function indicator() {
        return null;
      },
      ref: function ref(el) {
        return _this7.swiperRef = el;
      },
      defaultIndex: activeIndex,
      onIndexChange: function onIndexChange(index) {
        return _this7.onIndexChange(index);
      }
    }, dataSource.map(function (_ref4, key) {
      var recommendType = _ref4.recommendType;
      return /*#__PURE__*/_react["default"].createElement(_antdMobileV.Swiper.Item, {
        key: key
      }, key === activeIndex ? /*#__PURE__*/_react["default"].createElement("div", {
        className: "wrap",
        ref: function ref(el) {
          return _this7.masonryWrap = el;
        }
      }, !loading ? /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, listData.length !== 0 ? /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Masonry["default"], {
        dataSource: listData,
        col: recommendType === 7 ? 1 : 2,
        render: function render(data, index) {
          return /*#__PURE__*/_react["default"].createElement("div", {
            className: "card",
            key: index
          }, /*#__PURE__*/_react["default"].createElement(RenderContent, {
            data: data,
            recommendType: recommendType
          }));
        }
      }), /*#__PURE__*/_react["default"].createElement(_antdMobileV.InfiniteScroll, {
        style: {
          color: '#858B9C',
          fontSize: '0.26rem'
        },
        loadMore: function loadMore() {
          return _this7.loadMore();
        },
        hasMore: hasMore
      })) : /*#__PURE__*/_react["default"].createElement(_NoneConfig["default"], {
        type: 1
      })) : /*#__PURE__*/_react["default"].createElement(_SkeletonMasonry["default"], null)) : /*#__PURE__*/_react["default"].createElement("div", null));
    }))));
  };
  return MasonryLayout;
}(_react.Component);
MasonryLayout.defaultProps = {
  locationInfo: _const.defaultLocationInfo,
  appCode: _const.defaultAppCode,
  emptyDataVisible: false,
  attr: {
    name: '瀑布流'
  }
};
var _default = MasonryLayout;
exports["default"] = _default;
