// @umijs/plugin-request/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = _default;

function _react() {
  const data = _interopRequireDefault(require("react"));

  _react = function _react() {
    return data;
  };

  return data;
}

function _path() {
  const data = require("path");

  _path = function _path() {
    return data;
  };

  return data;
}

function _fs() {
  const data = require("fs");

  _fs = function _fs() {
    return data;
  };

  return data;
}

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _default(api) {
  const paths = api.paths,
        winPath = api.utils.winPath;
  api.addRuntimePluginKey(() => 'request');
  const umiRequestPkgPath = winPath((0, _path().dirname)(require.resolve('umi-request/package')));
  const useRequestPkgPath = winPath((0, _path().dirname)(require.resolve('@ahooksjs/use-request/package')));
  api.addDepInfo(() => {
    const pkg = require('../package.json');

    return [{
      name: 'umi-request',
      range: pkg.dependencies['umi-request'],
      alias: [umiRequestPkgPath]
    }, {
      name: '@ahooksjs/use-request',
      range: pkg.dependencies['@ahooksjs/use-request'],
      alias: [useRequestPkgPath]
    }];
  }); // 配置

  api.describe({
    config: {
      schema(joi) {
        return joi.object({
          dataField: joi.string().pattern(/^[a-zA-Z]*$/).allow('')
        });
      },

      default: {
        dataField: 'data'
      }
    }
  });
  const requestTemplate = (0, _fs().readFileSync)(winPath((0, _path().join)(__dirname, '../src/request.ts')), 'utf-8');
  const uiIndexTemplate = (0, _fs().readFileSync)(winPath((0, _path().join)(__dirname, '../src/ui/index.ts')), 'utf-8');
  const uiNoopTemplate = (0, _fs().readFileSync)(winPath((0, _path().join)(__dirname, '../src/ui/noop.ts')), 'utf-8');
  const namespace = 'plugin-request';
  api.chainWebpack(webpackConfig => {
    // decoupling antd ui library
    webpackConfig.resolve.alias.set('@umijs/plugin-request/lib/ui', api.config.antd === false ? winPath((0, _path().join)(api.paths.absTmpPath, namespace, './ui/noop')) : winPath((0, _path().join)(api.paths.absTmpPath, namespace, './ui/index')));
    return webpackConfig;
  });
  api.onGenerateFiles(() => {
    const _ref = api.config.request,
          _ref$dataField = _ref.dataField,
          dataField = _ref$dataField === void 0 ? 'data' : _ref$dataField;

    try {
      // Write .umi/plugin-request/request.ts
      let formatResultStr;

      if (dataField === '') {
        formatResultStr = 'formatResult: result => result';
      } else {
        formatResultStr = `formatResult: result => result?.${dataField}`;
      }

      api.writeTmpFile({
        path: winPath((0, _path().join)(namespace, 'request.ts')),
        content: requestTemplate.replace(/\/\*FRS\*\/(.+)\/\*FRE\*\//, formatResultStr).replace(/\['data'\]/g, dataField ? `['${dataField}']` : '').replace(/data\?: T;/, dataField ? `${dataField}?: T;` : '').replace(/umi-request/g, umiRequestPkgPath).replace(/@ahooksjs\/use-request/g, useRequestPkgPath).replace(`import { ApplyPluginsType, history, plugin } from 'umi';`, `
import { ApplyPluginsType } from 'umi';
import { history, plugin } from '../core/umiExports';
            `)
      });
      const uiTmpDir = (0, _path().join)(api.paths.absTmpPath, namespace, 'ui');
      !(0, _fs().existsSync)(uiTmpDir) && (0, _fs().mkdirSync)(uiTmpDir, {
        recursive: true
      });
      api.writeTmpFile({
        path: winPath((0, _path().join)(namespace, 'ui/index.ts')),
        content: uiIndexTemplate
      });
      api.writeTmpFile({
        path: winPath((0, _path().join)(namespace, 'ui/noop.ts')),
        content: uiNoopTemplate
      });
    } catch (e) {
      api.logger.error(e);
    }
  });
  api.addUmiExports(() => {
    return [{
      exportAll: true,
      source: `../${namespace}/request`
    }];
  });
}
