// @kuping/oto_component/lib/business/FooterNav/default.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.defaultDataSource = void 0;
var defaultDataSource = [{
  name: '标题',
  imgUrl: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670',
  pathUrl: '',
  activeStyle: {
    backgroundImage: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670'
  },
  style: {
    backgroundImage: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670'
  }
}, {
  name: '标题',
  imgUrl: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670',
  pathUrl: '',
  activeStyle: {
    backgroundImage: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670'
  },
  style: {
    backgroundImage: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670'
  }
}, {
  name: '标题',
  imgUrl: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670',
  pathUrl: '',
  activeStyle: {
    backgroundImage: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670'
  },
  style: {
    backgroundImage: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670'
  }
}, {
  name: '标题',
  imgUrl: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670',
  pathUrl: '',
  activeStyle: {
    backgroundImage: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670'
  },
  style: {
    backgroundImage: 'https://bosscdn.otosaas.com/test/20221108/20221108160322icon.png?path=0,650,670'
  }
}];
exports.defaultDataSource = defaultDataSource;
